{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "kPEJuKdp2o_G"
      },
      "source": [
        "# Leverage the DataCite REST API for metadata discovery and creation\n",
        "\n",
        "## Task Automation: Harvesting a set of metadata records\n",
        "\n",
        "### Overview\n",
        "\n",
        "Participants will learn how to:\n",
        "  1. Apply parameters to API requests using the Python requests library.\n",
        "  1. Create a loop to perform pagination, retrieving over 1,000 DOIs by making multiple requests.\n",
        "  1. Get results in different formats (JSON, CSV).\n",
        "  1. Save results to a file.\n",
        "\n",
        "\n",
        "* [DataCite Rest API Reference](https://support.datacite.org/reference/get_dois)\n",
        "* [Python Requests Package Reference](https://pypi.org/project/requests/)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "dOdzvn3zTbBL"
      },
      "source": [
        "### Applying parameters to API requests unsing `requests`"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "9yIXab84Y44u"
      },
      "source": [
        "To begin, we will import the requests library and set some constants representing the API endpoints"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "sfA3_bsxpr1Q"
      },
      "outputs": [],
      "source": [
        "import requests\n",
        "DATACITE_API_URL = \"https://api.datacite.org\"\n",
        "DATACITE_DOI_API_URL = f\"{DATACITE_API_URL}/dois/\"\n",
        "RESULTS_PER_PAGE = 1000"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "eDlmsOWtXvlp"
      },
      "source": [
        "At its simplest, the API can be used to fetch a query based on a string.\n",
        "We can constuct a url with query parameters in it"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "jG753E2BZz4-"
      },
      "outputs": [],
      "source": [
        "my_query = \"transcriptomes\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "Jw-g6LAPZNCp",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "380b847d-433c-4cd0-c87e-f9f5217cfec1"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "https://api.datacite.org/dois/?query=transcriptomes\n"
          ]
        }
      ],
      "source": [
        "url = DATACITE_DOI_API_URL + \"?\" + \"query=\" + my_query\n",
        "print(url)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "xkz_mnsHdlSi"
      },
      "source": [
        "We can then go out and fetch the data using a GET method"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "vM0TwRAeaxxG"
      },
      "outputs": [],
      "source": [
        "response = requests.get(url)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-gLZwV7Pd2X5"
      },
      "source": [
        "Assuming everything went alright we can convert the response to JSON"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "collapsed": true,
        "id": "h6irRkmybKth"
      },
      "outputs": [],
      "source": [
        "results = response.json()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "mb49qHYZu0vF"
      },
      "source": [
        "When we are checking things programatically, we need to explicitly check for errors.  This can be done with using the `response.status_code` or the shortcut `response.ok`"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VkfxVMf2v01u",
        "outputId": "d40a019f-1001-4593-c329-a55cd201fe03"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "200"
            ]
          },
          "metadata": {},
          "execution_count": 7
        }
      ],
      "source": [
        "response.status_code"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "t13z6O5WwJXl",
        "outputId": "0de31ed8-e656-4380-f792-7b107aa2bd68"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "True"
            ]
          },
          "metadata": {},
          "execution_count": 8
        }
      ],
      "source": [
        "response.ok"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "_Vr2KzozSFNb"
      },
      "source": [
        "You can also create a dictionary of parameters and pass this into the request that is sent to the API"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "eM2C-4EXSRcA"
      },
      "outputs": [],
      "source": [
        "rows = 10\n",
        "params = {\n",
        "        'query': my_query,\n",
        "        'page[size]': rows\n",
        "    }\n",
        "\n",
        "url = DATACITE_DOI_API_URL\n",
        "response = requests.get(url, params)\n",
        "if response.ok:\n",
        "  results = response.json()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "WLOXwV8rTZmG"
      },
      "source": [
        "Let's take a look at results and all results\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "frubSYjvUziT",
        "outputId": "cbe0f619-5966-4721-8749-130a77ff740e"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'data': [{'id': '10.7488/era/4551',\n",
              "   'type': 'dois',\n",
              "   'attributes': {'doi': '10.7488/era/4551',\n",
              "    'identifiers': [{'identifier': 'https://hdl.handle.net/1842/41828',\n",
              "      'identifierType': 'uri'}],\n",
              "    'creators': [{'name': 'Treanor-Taylor, Mairi',\n",
              "      'nameType': 'Personal',\n",
              "      'givenName': 'Mairi',\n",
              "      'familyName': 'Treanor-Taylor',\n",
              "      'affiliation': [],\n",
              "      'nameIdentifiers': []}],\n",
              "    'titles': [{'title': 'Investigating disease progression of cutaneous squamous cell carcinoma'}],\n",
              "    'publisher': 'The University of Edinburgh',\n",
              "    'container': {},\n",
              "    'publicationYear': 2024,\n",
              "    'subjects': [{'subject': 'cutaneous squamous cell carcinoma'},\n",
              "     {'subject': 'disease progression'},\n",
              "     {'subject': 'metastasis'},\n",
              "     {'subject': 'RNA-sequencing'},\n",
              "     {'subject': 'whole genome sequencing'}],\n",
              "    'contributors': [{'name': 'University Of Edinburgh',\n",
              "      'affiliation': [],\n",
              "      'contributorType': 'DataManager',\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'University Of Edinburgh',\n",
              "      'affiliation': [],\n",
              "      'contributorType': 'HostingInstitution',\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Gourley, Charlie',\n",
              "      'nameType': 'Personal',\n",
              "      'givenName': 'Charlie',\n",
              "      'familyName': 'Gourley',\n",
              "      'affiliation': [],\n",
              "      'contributorType': 'RelatedPerson',\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Bailey, Peter',\n",
              "      'nameType': 'Personal',\n",
              "      'givenName': 'Peter',\n",
              "      'familyName': 'Bailey',\n",
              "      'affiliation': [],\n",
              "      'contributorType': 'RelatedPerson',\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Inman, Gareth',\n",
              "      'nameType': 'Personal',\n",
              "      'givenName': 'Gareth',\n",
              "      'familyName': 'Inman',\n",
              "      'affiliation': [],\n",
              "      'contributorType': 'RelatedPerson',\n",
              "      'nameIdentifiers': []}],\n",
              "    'dates': [{'date': '2024-06-03', 'dateType': 'Accepted'},\n",
              "     {'date': '2024-06-03', 'dateType': 'Available'},\n",
              "     {'date': '2024-06-03', 'dateType': 'Issued'}],\n",
              "    'language': None,\n",
              "    'types': {'ris': 'GEN',\n",
              "     'bibtex': 'misc',\n",
              "     'citeproc': 'article',\n",
              "     'schemaOrg': 'CreativeWork'},\n",
              "    'relatedIdentifiers': [],\n",
              "    'relatedItems': [],\n",
              "    'sizes': [],\n",
              "    'formats': [],\n",
              "    'version': None,\n",
              "    'rightsList': [{'rights': '2025-06-03'}],\n",
              "    'descriptions': [{'description': 'Cutaneous squamous cell carcinoma (cSCC) is the most common type of potentially lethal cancer. cSCC has a good prognosis when caught early, yet 2-5% metastasise, where they are as fatal as any other metastatic cancer. Due to the high prevalence and relatively low metastasis rate, cSCC are often omitted from major cancer registries. The subsequent lack of research into this cancer type has meant that the molecular mechanisms that drive disease progression from pre-malignant actinic keratosis (AK) to malignant cSCC and their associated potential therapeutic vulnerabilities remain to be defined. To investigate these processes, we performed transcriptomic profiling of 113 patient samples including normal sun exposed skin (n=29), AK (n=14), primary cSCC (n=66) and metastases (n=4). Using gene expression signature analysis, we revealed that cSCC disease progression can be represented as a continuum from a differentiated to a progenitor like state (DP signature). Dividing samples into quartiles based on DP signature score indicated that samples in the most progenitor like state (quartile 1) are associated with increased tumour size, depth of invasion and pathologist defined poor differentiation status. To identify genes that may drive disease progression we employed differential gene expression analysis to identify genes upregulated along disease progression, in poorly differentiated tumours, and in DP quartile 1. We further filtered candidate drivers based on expression levels in a panel of patient derived cSCC cell lines (PDCLs) and investigated the potential tumour cell autonomous role of 175 genes in a high-content siRNA pool knockdown screen. 64 genes were selected for siRNA deconvolution analysis and further high-content screening assessing cell proliferation, clustering, and morphology. These studies identified 9 candidate genes as positive regulators of cSCC cell proliferation, which we are investigating further. Whole genome sequencing data from primary and metastatic samples from three patients were analysed as an alternative approach to investigate metastatic cSCC. Loss of TGF\\uf062R2 and activation of PI3 Kinase signalling was associated with metastasis and was collaboratively functionally validated using genetically engineered mouse models of cSCC. To further investigate the biological basis of metastasis we transcriptionally profiled perilesional normal tissue, primary cSCC tumour tissue and matched metastases from 224 immunocompetent patients (140 non-metastasising and 84 metastasising). Bioinformatic analyses of these transcriptomes revealed pathways, processes and genes that may be important in driving metastasis.',\n",
              "      'descriptionType': 'Abstract'}],\n",
              "    'geoLocations': [],\n",
              "    'fundingReferences': [],\n",
              "    'url': 'https://era.ed.ac.uk/handle/1842/41828',\n",
              "    'contentUrl': None,\n",
              "    'metadataVersion': 0,\n",
              "    'schemaVersion': 'http://datacite.org/schema/kernel-3',\n",
              "    'source': 'mds',\n",
              "    'isActive': True,\n",
              "    'state': 'findable',\n",
              "    'reason': None,\n",
              "    'viewCount': 0,\n",
              "    'downloadCount': 0,\n",
              "    'referenceCount': 0,\n",
              "    'citationCount': 0,\n",
              "    'partCount': 0,\n",
              "    'partOfCount': 0,\n",
              "    'versionCount': 0,\n",
              "    'versionOfCount': 0,\n",
              "    'created': '2024-06-03T10:00:11Z',\n",
              "    'registered': '2024-06-03T10:00:23Z',\n",
              "    'published': None,\n",
              "    'updated': '2024-06-03T14:00:13Z'},\n",
              "   'relationships': {'client': {'data': {'id': 'bl.ed', 'type': 'clients'}}}},\n",
              "  {'id': '10.60692/7mnfs-10y79',\n",
              "   'type': 'dois',\n",
              "   'attributes': {'doi': '10.60692/7mnfs-10y79',\n",
              "    'identifiers': [{'identifier': 'oai:gresis.org:7mnfs-10y79',\n",
              "      'identifierType': 'oai'},\n",
              "     {'identifier': 'https://openalex.org/W4388018330',\n",
              "      'identifierType': 'Other'}],\n",
              "    'creators': [{'name': 'Sunita Yadav',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Sunita Yadav',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0003-2125-183X',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Priyanka Mehta',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Priyanka Mehta',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0001-6298-4322',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Jyoti Soni',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Jyoti Soni',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Partha Chattopadhyay',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Partha Chattopadhyay',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-5536-3162',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Priti Devi',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Priti Devi',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Thierry Habyarimana',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Thierry Habyarimana',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'INES-Ruhengeri'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0001-5488-5578',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Kishore Tardalkar',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Kishore Tardalkar',\n",
              "      'affiliation': ['Dr. D. Y. Patil Medical College, Hospital and Research Centre'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Meghnad Joshi',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Meghnad Joshi',\n",
              "      'affiliation': ['Dr. D. Y. Patil Medical College, Hospital and Research Centre'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0001-6060-244X',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Rajesh Pandey',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Rajesh Pandey',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-4404-8327',\n",
              "        'nameIdentifierScheme': 'ORCID'}]}],\n",
              "    'titles': [{'title': 'Single-cell RNA-Seq reveals intracellular microbial diversity within immune cells during SARS-CoV-2 infection and recovery'},\n",
              "     {'lang': 'ara',\n",
              "      'title': 'يكشف الحمض النووي الريبي أحادي الخلية عن التنوع الميكروبي داخل الخلايا المناعية أثناء عدوى فيروس كورونا 2 المرتبط بمتلازمة الجهاز التنفسي الحادة الوخيمة والشفاء منه',\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'fra',\n",
              "      'title': \"L'ARN-Seq unicellulaire révèle la diversité microbienne intracellulaire au sein des cellules immunitaires pendant l'infection par le SRAS-CoV-2 et la récupération\",\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'spa',\n",
              "      'title': 'La ARN-Seq unicelular revela la diversidad microbiana intracelular dentro de las células inmunitarias durante la infección y recuperación del SARS-CoV-2',\n",
              "      'titleType': 'TranslatedTitle'}],\n",
              "    'publisher': 'OpenAlex',\n",
              "    'container': {},\n",
              "    'publicationYear': 2023,\n",
              "    'subjects': [{'subject': 'Neurological Manifestations of COVID-19 Infection'},\n",
              "     {'subject': 'Neurology'},\n",
              "     {'subject': 'Medicine'},\n",
              "     {'subject': 'Health Sciences'},\n",
              "     {'subject': 'Biology'},\n",
              "     {'subject': 'Microbiology'},\n",
              "     {'subject': 'FOS: Biological sciences',\n",
              "      'schemeUri': 'http://www.oecd.org/science/inno/38235147.pdf',\n",
              "      'subjectScheme': 'Fields of Science and Technology (FOS)'},\n",
              "     {'subject': 'Transcriptome'},\n",
              "     {'subject': 'Microorganism'},\n",
              "     {'subject': 'Immune system'},\n",
              "     {'subject': 'Intracellular'},\n",
              "     {'subject': 'Single-cell analysis'},\n",
              "     {'subject': 'Prokaryote'},\n",
              "     {'subject': 'Intracellular parasite'},\n",
              "     {'subject': 'Bacteria'},\n",
              "     {'subject': 'Cell'},\n",
              "     {'subject': 'Genetics'},\n",
              "     {'subject': 'Gene expression'},\n",
              "     {'subject': 'Gene'}],\n",
              "    'contributors': [],\n",
              "    'dates': [{'date': '2023-11-01', 'dateType': 'Issued'}],\n",
              "    'language': 'en',\n",
              "    'types': {'ris': 'RPRT',\n",
              "     'bibtex': 'article',\n",
              "     'citeproc': 'article-journal',\n",
              "     'schemaOrg': 'ScholarlyArticle',\n",
              "     'resourceType': '',\n",
              "     'resourceTypeGeneral': 'Text'},\n",
              "    'relatedIdentifiers': [{'relationType': 'IsVersionOf',\n",
              "      'relatedIdentifier': '10.60692/k1t8q-38c08',\n",
              "      'relatedIdentifierType': 'DOI'}],\n",
              "    'relatedItems': [],\n",
              "    'sizes': [],\n",
              "    'formats': [],\n",
              "    'version': 'v1',\n",
              "    'rightsList': [{'rights': 'cc-by'}],\n",
              "    'descriptions': [{'description': \"Intracellular microorganisms, like viruses, bacteria, and fungi, pose challenges in detection due to their non-culturable forms. Transcriptomic analysis at cellular level enables exploration of distributions and the impact of these microorganisms on host cells, a domain that remains underexplored because of methodological limitations. Single-cell technology shows promise in addressing this by capturing polyadenine-tailed transcripts, because recent studies confirmed polyadenylation in microbial transcriptomes. We utilized single-cell RNA-seq from PBMCs to probe intracellular microbes in healthy, SARS-CoV-2-positive, and recovered individuals. Among 76 bacterial species detected, 16 showed significant abundance differences. Buchnera aphidicola, Streptomyces clavuligerus, and Ehrlichia canis emerged significantly in memory-B, Naïve-T, and Treg cells. Staphylococcus aureus, Mycoplasma mycoides, Leptospira interrogans, and others displayed elevated levels in SARS-CoV-2-positive patients, suggesting possible disease association. This highlights the strength of single-cell technology in revealing potential microorganism's cell-specific functions. Further research is essential for functional understanding of their cell-specific abundance across physiological states.\",\n",
              "      'descriptionType': 'Abstract'},\n",
              "     {'lang': 'ara',\n",
              "      'description': 'تشكل الكائنات الحية الدقيقة داخل الخلايا، مثل الفيروسات والبكتيريا والفطريات، تحديات في الكشف بسبب أشكالها غير القابلة للزراعة. يتيح التحليل النصي على المستوى الخلوي استكشاف التوزيعات وتأثير هذه الكائنات الحية الدقيقة على الخلايا المضيفة، وهو مجال لا يزال غير مستكشف بسبب القيود المنهجية. تُظهر التكنولوجيا أحادية الخلية وعداً في معالجة هذا من خلال التقاط النصوص متعددة الأدينين، لأن الدراسات الحديثة أكدت تعدد الأدينيل في النسخ الميكروبية. استخدمنا تسلسل الحمض النووي الريبي أحادي الخلية من PBMCs لفحص الميكروبات داخل الخلايا في الأفراد الأصحاء والإيجابيين لفيروس كورونا 2 المرتبط بمتلازمة الجهاز التنفسي الحادة الوخيمة (SARS - CoV -2) والمتعافين. من بين 76 نوعًا بكتيريًا تم اكتشافها، أظهر 16 نوعًا اختلافات كبيرة في الوفرة. ظهرت كل من Buchnera aphidicola و Streptomyces clavuligerus و Ehrlichia canis بشكل ملحوظ في خلايا الذاكرة B و Naïve - T و Treg. أظهرت المكورات العنقودية الذهبية، والميكوبلازما الفطرية، والبريميات الاستجوابية، وغيرها مستويات مرتفعة في المرضى المصابين بفيروس كورونا 2 المرتبط بمتلازمة الجهاز التنفسي الحادة الوخيمة، مما يشير إلى ارتباط محتمل بالمرض. وهذا يسلط الضوء على قوة تقنية الخلية الواحدة في الكشف عن وظائف الخلايا الخاصة بالكائنات الحية الدقيقة المحتملة. يعد إجراء المزيد من الأبحاث أمرًا ضروريًا للفهم الوظيفي لوفرة الخلايا الخاصة بها عبر الحالات الفسيولوجية.',\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'fra',\n",
              "      'description': \"Les micro-organismes intracellulaires, comme les virus, les bactéries et les champignons, posent des problèmes de détection en raison de leurs formes non cultivables. L'analyse transcriptomique au niveau cellulaire permet d'explorer les distributions et l'impact de ces microorganismes sur les cellules hôtes, domaine qui reste sous-exploré en raison des limites méthodologiques. La technologie unicellulaire est prometteuse pour résoudre ce problème en capturant les transcrits à queue de polyadénine, car des études récentes ont confirmé la polyadénylation dans les transcriptomes microbiens. Nous avons utilisé une séquence d'ARN unicellulaire provenant de PBMC pour sonder les microbes intracellulaires chez des individus sains, positifs au SRAS-CoV-2 et rétablis. Parmi les 76 espèces bactériennes détectées, 16 ont montré des différences d'abondance significatives. Buchnera aphidicola, Streptomyces clavuligerus et Ehrlichia canis ont émergé de manière significative dans les cellules Memory-B, Naïve-T et Treg. Staphylococcus aureus, Mycoplasma mycoides, Leptospira interrogans et d'autres ont montré des niveaux élevés chez les patients positifs au SRAS-CoV-2, suggérant une association possible avec la maladie. Cela met en évidence la force de la technologie unicellulaire pour révéler les fonctions spécifiques aux cellules des micro-organismes potentiels. Des recherches supplémentaires sont essentielles pour la compréhension fonctionnelle de leur abondance spécifique aux cellules dans tous les états physiologiques.\",\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'spa',\n",
              "      'description': 'Los microorganismos intracelulares, como virus, bacterias y hongos, plantean desafíos en la detección debido a sus formas no cultivables. El análisis transcriptómico a nivel celular permite explorar las distribuciones y el impacto de estos microorganismos en las células huésped, un dominio que permanece poco explorado debido a las limitaciones metodológicas. La tecnología de una sola célula es prometedora para abordar esto mediante la captura de transcritos con cola de poliadenina, porque estudios recientes confirmaron la poliadenilación en transcriptomas microbianos. Utilizamos ARN-seq unicelular de PBMC para sondear microbios intracelulares en individuos sanos, SARS-CoV-2 positivos y recuperados. Entre 76 especies bacterianas detectadas, 16 mostraron diferencias de abundancia significativas. Buchnera aphidicola, Streptomyces clavuligerus y Ehrlichia canis surgieron significativamente en las células Memory-B, Naïve-T y Treg. Staphylococcus aureus, Mycoplasma mycoides, Leptospira interrogans y otros mostraron niveles elevados en pacientes positivos para SARS-CoV-2, lo que sugiere una posible asociación con la enfermedad. Esto destaca la fortaleza de la tecnología unicelular para revelar las posibles funciones específicas de las células de los microorganismos. La investigación adicional es esencial para la comprensión funcional de su abundancia específica de células en todos los estados fisiológicos.',\n",
              "      'descriptionType': 'Other'}],\n",
              "    'geoLocations': [],\n",
              "    'fundingReferences': [],\n",
              "    'url': 'https://gresis.osc.int//doi/10.60692/7mnfs-10y79',\n",
              "    'contentUrl': None,\n",
              "    'metadataVersion': 0,\n",
              "    'schemaVersion': 'http://datacite.org/schema/kernel-4',\n",
              "    'source': 'api',\n",
              "    'isActive': True,\n",
              "    'state': 'findable',\n",
              "    'reason': None,\n",
              "    'viewCount': 0,\n",
              "    'downloadCount': 0,\n",
              "    'referenceCount': 0,\n",
              "    'citationCount': 0,\n",
              "    'partCount': 0,\n",
              "    'partOfCount': 0,\n",
              "    'versionCount': 0,\n",
              "    'versionOfCount': 0,\n",
              "    'created': '2024-06-03T12:18:21Z',\n",
              "    'registered': '2024-06-03T12:18:21Z',\n",
              "    'published': None,\n",
              "    'updated': '2024-06-03T12:18:21Z'},\n",
              "   'relationships': {'client': {'data': {'id': 'iapx.bsycxq',\n",
              "      'type': 'clients'}}}},\n",
              "  {'id': '10.60692/k1t8q-38c08',\n",
              "   'type': 'dois',\n",
              "   'attributes': {'doi': '10.60692/k1t8q-38c08',\n",
              "    'identifiers': [{'identifier': 'https://openalex.org/W4388018330',\n",
              "      'identifierType': 'Other'}],\n",
              "    'creators': [{'name': 'Sunita Yadav',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Sunita Yadav',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0003-2125-183X',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Priyanka Mehta',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Priyanka Mehta',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0001-6298-4322',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Jyoti Soni',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Jyoti Soni',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Partha Chattopadhyay',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Partha Chattopadhyay',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-5536-3162',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Priti Devi',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Priti Devi',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Thierry Habyarimana',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Thierry Habyarimana',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'INES-Ruhengeri'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0001-5488-5578',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Kishore Tardalkar',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Kishore Tardalkar',\n",
              "      'affiliation': ['Dr. D. Y. Patil Medical College, Hospital and Research Centre'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Meghnad Joshi',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Meghnad Joshi',\n",
              "      'affiliation': ['Dr. D. Y. Patil Medical College, Hospital and Research Centre'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0001-6060-244X',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Rajesh Pandey',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Rajesh Pandey',\n",
              "      'affiliation': ['Institute of Genomics and Integrative Biology',\n",
              "       'Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-4404-8327',\n",
              "        'nameIdentifierScheme': 'ORCID'}]}],\n",
              "    'titles': [{'title': 'Single-cell RNA-Seq reveals intracellular microbial diversity within immune cells during SARS-CoV-2 infection and recovery'},\n",
              "     {'lang': 'ara',\n",
              "      'title': 'يكشف الحمض النووي الريبي أحادي الخلية عن التنوع الميكروبي داخل الخلايا المناعية أثناء عدوى فيروس كورونا 2 المرتبط بمتلازمة الجهاز التنفسي الحادة الوخيمة والشفاء منه',\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'fra',\n",
              "      'title': \"L'ARN-Seq unicellulaire révèle la diversité microbienne intracellulaire au sein des cellules immunitaires pendant l'infection par le SRAS-CoV-2 et la récupération\",\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'spa',\n",
              "      'title': 'La ARN-Seq unicelular revela la diversidad microbiana intracelular dentro de las células inmunitarias durante la infección y recuperación del SARS-CoV-2',\n",
              "      'titleType': 'TranslatedTitle'}],\n",
              "    'publisher': 'OpenAlex',\n",
              "    'container': {},\n",
              "    'publicationYear': 2023,\n",
              "    'subjects': [{'subject': 'Neurological Manifestations of COVID-19 Infection'},\n",
              "     {'subject': 'Neurology'},\n",
              "     {'subject': 'Medicine'},\n",
              "     {'subject': 'Health Sciences'},\n",
              "     {'subject': 'Biology'},\n",
              "     {'subject': 'Microbiology'},\n",
              "     {'subject': 'FOS: Biological sciences',\n",
              "      'schemeUri': 'http://www.oecd.org/science/inno/38235147.pdf',\n",
              "      'subjectScheme': 'Fields of Science and Technology (FOS)'},\n",
              "     {'subject': 'Transcriptome'},\n",
              "     {'subject': 'Microorganism'},\n",
              "     {'subject': 'Immune system'},\n",
              "     {'subject': 'Intracellular'},\n",
              "     {'subject': 'Single-cell analysis'},\n",
              "     {'subject': 'Prokaryote'},\n",
              "     {'subject': 'Intracellular parasite'},\n",
              "     {'subject': 'Bacteria'},\n",
              "     {'subject': 'Cell'},\n",
              "     {'subject': 'Genetics'},\n",
              "     {'subject': 'Gene expression'},\n",
              "     {'subject': 'Gene'}],\n",
              "    'contributors': [],\n",
              "    'dates': [{'date': '2023-11-01', 'dateType': 'Issued'}],\n",
              "    'language': 'en',\n",
              "    'types': {'ris': 'RPRT',\n",
              "     'bibtex': 'article',\n",
              "     'citeproc': 'article-journal',\n",
              "     'schemaOrg': 'ScholarlyArticle',\n",
              "     'resourceType': '',\n",
              "     'resourceTypeGeneral': 'Text'},\n",
              "    'relatedIdentifiers': [{'relationType': 'HasVersion',\n",
              "      'relatedIdentifier': '10.60692/7mnfs-10y79',\n",
              "      'relatedIdentifierType': 'DOI'}],\n",
              "    'relatedItems': [],\n",
              "    'sizes': [],\n",
              "    'formats': [],\n",
              "    'version': 'v1',\n",
              "    'rightsList': [{'rights': 'cc-by'}],\n",
              "    'descriptions': [{'description': \"Intracellular microorganisms, like viruses, bacteria, and fungi, pose challenges in detection due to their non-culturable forms. Transcriptomic analysis at cellular level enables exploration of distributions and the impact of these microorganisms on host cells, a domain that remains underexplored because of methodological limitations. Single-cell technology shows promise in addressing this by capturing polyadenine-tailed transcripts, because recent studies confirmed polyadenylation in microbial transcriptomes. We utilized single-cell RNA-seq from PBMCs to probe intracellular microbes in healthy, SARS-CoV-2-positive, and recovered individuals. Among 76 bacterial species detected, 16 showed significant abundance differences. Buchnera aphidicola, Streptomyces clavuligerus, and Ehrlichia canis emerged significantly in memory-B, Naïve-T, and Treg cells. Staphylococcus aureus, Mycoplasma mycoides, Leptospira interrogans, and others displayed elevated levels in SARS-CoV-2-positive patients, suggesting possible disease association. This highlights the strength of single-cell technology in revealing potential microorganism's cell-specific functions. Further research is essential for functional understanding of their cell-specific abundance across physiological states.\",\n",
              "      'descriptionType': 'Abstract'},\n",
              "     {'lang': 'ara',\n",
              "      'description': 'تشكل الكائنات الحية الدقيقة داخل الخلايا، مثل الفيروسات والبكتيريا والفطريات، تحديات في الكشف بسبب أشكالها غير القابلة للزراعة. يتيح التحليل النصي على المستوى الخلوي استكشاف التوزيعات وتأثير هذه الكائنات الحية الدقيقة على الخلايا المضيفة، وهو مجال لا يزال غير مستكشف بسبب القيود المنهجية. تُظهر التكنولوجيا أحادية الخلية وعداً في معالجة هذا من خلال التقاط النصوص متعددة الأدينين، لأن الدراسات الحديثة أكدت تعدد الأدينيل في النسخ الميكروبية. استخدمنا تسلسل الحمض النووي الريبي أحادي الخلية من PBMCs لفحص الميكروبات داخل الخلايا في الأفراد الأصحاء والإيجابيين لفيروس كورونا 2 المرتبط بمتلازمة الجهاز التنفسي الحادة الوخيمة (SARS - CoV -2) والمتعافين. من بين 76 نوعًا بكتيريًا تم اكتشافها، أظهر 16 نوعًا اختلافات كبيرة في الوفرة. ظهرت كل من Buchnera aphidicola و Streptomyces clavuligerus و Ehrlichia canis بشكل ملحوظ في خلايا الذاكرة B و Naïve - T و Treg. أظهرت المكورات العنقودية الذهبية، والميكوبلازما الفطرية، والبريميات الاستجوابية، وغيرها مستويات مرتفعة في المرضى المصابين بفيروس كورونا 2 المرتبط بمتلازمة الجهاز التنفسي الحادة الوخيمة، مما يشير إلى ارتباط محتمل بالمرض. وهذا يسلط الضوء على قوة تقنية الخلية الواحدة في الكشف عن وظائف الخلايا الخاصة بالكائنات الحية الدقيقة المحتملة. يعد إجراء المزيد من الأبحاث أمرًا ضروريًا للفهم الوظيفي لوفرة الخلايا الخاصة بها عبر الحالات الفسيولوجية.',\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'fra',\n",
              "      'description': \"Les micro-organismes intracellulaires, comme les virus, les bactéries et les champignons, posent des problèmes de détection en raison de leurs formes non cultivables. L'analyse transcriptomique au niveau cellulaire permet d'explorer les distributions et l'impact de ces microorganismes sur les cellules hôtes, domaine qui reste sous-exploré en raison des limites méthodologiques. La technologie unicellulaire est prometteuse pour résoudre ce problème en capturant les transcrits à queue de polyadénine, car des études récentes ont confirmé la polyadénylation dans les transcriptomes microbiens. Nous avons utilisé une séquence d'ARN unicellulaire provenant de PBMC pour sonder les microbes intracellulaires chez des individus sains, positifs au SRAS-CoV-2 et rétablis. Parmi les 76 espèces bactériennes détectées, 16 ont montré des différences d'abondance significatives. Buchnera aphidicola, Streptomyces clavuligerus et Ehrlichia canis ont émergé de manière significative dans les cellules Memory-B, Naïve-T et Treg. Staphylococcus aureus, Mycoplasma mycoides, Leptospira interrogans et d'autres ont montré des niveaux élevés chez les patients positifs au SRAS-CoV-2, suggérant une association possible avec la maladie. Cela met en évidence la force de la technologie unicellulaire pour révéler les fonctions spécifiques aux cellules des micro-organismes potentiels. Des recherches supplémentaires sont essentielles pour la compréhension fonctionnelle de leur abondance spécifique aux cellules dans tous les états physiologiques.\",\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'spa',\n",
              "      'description': 'Los microorganismos intracelulares, como virus, bacterias y hongos, plantean desafíos en la detección debido a sus formas no cultivables. El análisis transcriptómico a nivel celular permite explorar las distribuciones y el impacto de estos microorganismos en las células huésped, un dominio que permanece poco explorado debido a las limitaciones metodológicas. La tecnología de una sola célula es prometedora para abordar esto mediante la captura de transcritos con cola de poliadenina, porque estudios recientes confirmaron la poliadenilación en transcriptomas microbianos. Utilizamos ARN-seq unicelular de PBMC para sondear microbios intracelulares en individuos sanos, SARS-CoV-2 positivos y recuperados. Entre 76 especies bacterianas detectadas, 16 mostraron diferencias de abundancia significativas. Buchnera aphidicola, Streptomyces clavuligerus y Ehrlichia canis surgieron significativamente en las células Memory-B, Naïve-T y Treg. Staphylococcus aureus, Mycoplasma mycoides, Leptospira interrogans y otros mostraron niveles elevados en pacientes positivos para SARS-CoV-2, lo que sugiere una posible asociación con la enfermedad. Esto destaca la fortaleza de la tecnología unicelular para revelar las posibles funciones específicas de las células de los microorganismos. La investigación adicional es esencial para la comprensión funcional de su abundancia específica de células en todos los estados fisiológicos.',\n",
              "      'descriptionType': 'Other'}],\n",
              "    'geoLocations': [],\n",
              "    'fundingReferences': [],\n",
              "    'url': 'https://gresis.osc.int//doi/10.60692/k1t8q-38c08',\n",
              "    'contentUrl': None,\n",
              "    'metadataVersion': 0,\n",
              "    'schemaVersion': 'http://datacite.org/schema/kernel-4',\n",
              "    'source': 'api',\n",
              "    'isActive': True,\n",
              "    'state': 'findable',\n",
              "    'reason': None,\n",
              "    'viewCount': 0,\n",
              "    'downloadCount': 0,\n",
              "    'referenceCount': 0,\n",
              "    'citationCount': 0,\n",
              "    'partCount': 0,\n",
              "    'partOfCount': 0,\n",
              "    'versionCount': 0,\n",
              "    'versionOfCount': 0,\n",
              "    'created': '2024-06-03T12:18:21Z',\n",
              "    'registered': '2024-06-03T12:18:21Z',\n",
              "    'published': None,\n",
              "    'updated': '2024-06-03T12:18:21Z'},\n",
              "   'relationships': {'client': {'data': {'id': 'iapx.bsycxq',\n",
              "      'type': 'clients'}}}},\n",
              "  {'id': '10.5281/zenodo.3702901',\n",
              "   'type': 'dois',\n",
              "   'attributes': {'doi': '10.5281/zenodo.3702901',\n",
              "    'identifiers': [{'identifier': 'oai:zenodo.org:3702901',\n",
              "      'identifierType': 'oai'}],\n",
              "    'creators': [{'name': 'Craig McCormick',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Craig McCormick',\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0003-2794-3722',\n",
              "        'nameIdentifierScheme': 'ORCID'}],\n",
              "      'affiliation': []}],\n",
              "    'titles': [{'title': 'Review of In Vivo Dual RNA-Seq Analysis Reveals the Basis for Differential Tissue Tropism of Clinical Isolates of Streptococcus pneumoniae'}],\n",
              "    'publisher': 'Zenodo',\n",
              "    'container': {'type': 'Series',\n",
              "     'identifier': 'https://zenodo.org/communities/prereview-reviews',\n",
              "     'identifierType': 'URL'},\n",
              "    'publicationYear': 2020,\n",
              "    'subjects': [],\n",
              "    'contributors': [],\n",
              "    'dates': [{'date': '2020-03-10', 'dateType': 'Issued'}],\n",
              "    'language': 'en',\n",
              "    'types': {'ris': 'GEN',\n",
              "     'bibtex': 'misc',\n",
              "     'citeproc': 'article',\n",
              "     'schemaOrg': 'Review',\n",
              "     'resourceType': '',\n",
              "     'resourceTypeGeneral': 'PeerReview'},\n",
              "    'relatedIdentifiers': [{'relationType': 'Reviews',\n",
              "      'relatedIdentifier': '10.1101/862755',\n",
              "      'resourceTypeGeneral': 'Preprint',\n",
              "      'relatedIdentifierType': 'DOI'},\n",
              "     {'relationType': 'IsIdenticalTo',\n",
              "      'relatedIdentifier': 'https://prereview.org/reviews/3702901',\n",
              "      'resourceTypeGeneral': 'PeerReview',\n",
              "      'relatedIdentifierType': 'URL'},\n",
              "     {'relationType': 'IsVersionOf',\n",
              "      'relatedIdentifier': '10.5281/zenodo.3702900',\n",
              "      'relatedIdentifierType': 'DOI'}],\n",
              "    'relatedItems': [],\n",
              "    'sizes': [],\n",
              "    'formats': [],\n",
              "    'version': None,\n",
              "    'rightsList': [{'rights': 'Creative Commons Attribution 4.0 International',\n",
              "      'rightsUri': 'https://creativecommons.org/licenses/by/4.0/legalcode',\n",
              "      'schemeUri': 'https://spdx.org/licenses/',\n",
              "      'rightsIdentifier': 'cc-by-4.0',\n",
              "      'rightsIdentifierScheme': 'SPDX'}],\n",
              "    'descriptions': [{'description': 'This Zenodo record is a permanently preserved version of a PREreview. You can view the complete PREreview at https://prereview.org/reviews/3702901.\\n\\n\\nWe, the students of MICI5029/5049, a Graduate Level Molecular Pathogenesis Journal Club at Dalhousie University in Halifax, NS, Canada, hereby submit a review of the following\\xa0BioRxiv\\xa0preprint:\\n\\n\\nVikrant\\xa0Minhas,\\xa0Rieza\\xa0Aprianto,\\xa0Lauren J.\\xa0McAllister,\\xa0Hui\\xa0Wang,\\xa0Shannon C.\\xa0David,\\xa0Kimberley T.\\xa0McLean,\\xa0Iain\\xa0Comerford,\\xa0Shaun R.\\xa0McColl,\\xa0James C.\\xa0Paton,\\xa0Jan-Willem\\xa0Veening,\\xa0Claudia\\xa0Trappetti. In Vivo\\xa0Dual RNA-Seq Analysis Reveals the Basis for Differential Tissue Tropism of Clinical Isolates of\\xa0Streptococcus pneumoniae.\\xa0BioRxiv\\xa0862755; doi:\\xa0https://doi.org/10.1101/862755\\n\\n\\n\\xa0\\n\\n\\nWe will adhere to the Universal Principled (UP) Review guidelines proposed in:\\n\\n\\n\\xa0\\n\\n\\nUniversal Principled Review: A Community-Driven Method to Improve Peer Review.\\xa0Krummel M, Blish C, Kuhns M, Cadwell K, Oberst A, Goldrath A, Ansel KM, Chi H, O\\'Connell R, Wherry EJ, Pepper M; Future Immunology Consortium.\\xa0Cell. 2019 Dec 12;179(7):1441-1445.\\xa0https://doi.org/10.1016/j.cell.2019.11.029\\n\\n\\n\\xa0\\n\\n\\nSUMMARY:\\xa0Previous work has shown that two genetically related clinical isolates (947-Ear RafRD249\\xa0isolate and 4559-Blood RafRG249\\xa0isolate) of\\xa0Streptococcus pneumoniae\\xa0serotype 14 ST15 display differences in virulence in the mouse lung following intranasal challenge. These different virulence phenotypes of the two clinical isolates correlated with non-conserved single nucleotide polymorphisms (SNPs) in the raffinose pathway regulatory gene\\xa0rafR\\xa0(RafRD249G). In this study, Minhas\\xa0et al. investigated the effects of the RafRD249G\\xa0SNP on bacterial and host transcriptomes in infected lungs using a dual RNA-seq approach. They observed differential expression of genes encoding sugar transporters\\xa0that fine-tune carbohydrate metabolism, which were driven by the RafRD249G\\xa0SNP. This contributed to distinct host niche specialization for the two clinical isolates. RNA-seq analysis of the host response to the\\xa0S.\\xa0pneumoniae\\xa0infection\\xa0suggested that the RafRD249G\\xa0SNP in bacteria caused differential expression of host genes encoding for multiple cytokines, cytokine receptors, chemokines and chemokine ligands. Particularly, the expression of IL-17-related genes was enriched in the murine lungs infected with the 947-Ear isolate carrying the RafRD249\\xa0SNP compared to the 947-Ear isolate carrying the RafRG249\\xa0SNP. Using\\xa0in vivo\\xa0neutrophil depletion and IL-17A neutralization, the authors found that IL-17-induced neutrophil recruitment was partially responsible for the more efficient clearance of isolates of 947-Ear RafRD249\\xa0observed in the murine lungs. Overall, this study showed that the RafRD249G\\xa0SNP had a significant impact on bacterial and host transcriptomes, leading to distinct virulence phenotypes and host response. The use of dual RNA-seq in this study provided a powerful approach for investigating\\xa0in vivo\\xa0host-pathogen interactions.\\n\\n\\n\\xa0\\n\\n\\nOVERALL ASSESSMENT:\\xa0\\n\\n\\n\\xa0\\n\\n\\nSTRENGTHS: The authors conducted a transcriptomic study on the impact of a non-conservative SNPs RafRD249G\\xa0on host cells and two\\xa0S. pneumoniae\\xa0isolates (ear and blood) using a dual RNA-seq approach in murine lung infection. Their RNA-seq data strongly suggested that the RafRD249G\\xa0SNP between the two isolates caused distinct patterns of gene expression. Moreover, the dual RNA-seq on the host side showed that the RafRD249G\\xa0SNP also played a detrimental role in trigging differential host responses in infected lung. This study provides solid evidence that the dual RNA-seq approach is a useful tool for\\xa0in vivo\\xa0study of complex host-pathogen interactions.\\xa0\\n\\n\\n\\xa0\\n\\n\\nWEAKNESSES: There are a few weaknesses in the data analysis, data presentation, and experimental design. The lack of information in RNA-seq data analysis made it difficult for the reviewers to evaluate data quality. Some of the figure labels are difficult to interpret. In certain instances, methods were not described in sufficient detail to enable others to reproduce experiments. In this study, the RNA-seq data suggested an association between IL-17 expression and the RafRD249G\\xa0SNP in bacteria. However, the conclusion that RafRD249G\\xa0SNP was responsible for bacterial clearance via IL-17-mediated neutrophil recruitment was insufficiently supported. This could be improved by using appropriate controls, testing neutrophil recruitment in murine lung, and comparing data within each treatment. Moreover, we found that some of the writing in the manuscript was misleading and confusing. We believe that some of the weaknesses mentioned above could be solved by simplifying and clarifying the data presentation and interpretation.\\n\\n\\n\\xa0\\n\\n\\nDETAILED U.P. ASSESSMENT:\\n\\n\\n\\xa0\\n\\n\\nOBJECTIVE CRITERIA (QUALITY):\\xa0\\n\\n\\n\\xa0\\n\\n\\n1.\\xa0\\xa0\\xa0\\xa0\\xa0Quality: Experiments (1-3 scale) SCORE =\\xa01.5\\n\\n\\nFigure by Figure, do experiments, as performed, have the proper controls?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Figures 2B and 2C: There appears to be an enrichment in higher fold changes on either side of the replication origin. Have the authors considered whether cell populations might be rapidly growing, which could provide multiple copies of the origin per cell, thereby clouding interpretation?\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Figure 5: The gating strategy for multi-color flow cytometry analysis should be provided.\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Figure 5 and 6A: We think adding a mock infected negative control or showing the data of time zero as the base line is necessary to interpret this data.\\xa0\\n\\n\\n\\xa0\\n\\n\\nAre specific analyses performed using methods that are consistent with answering the specific question? Is there the appropriate technical expertise in the collection and analysis of data presented?\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In Figure 2&3, the meaning of \"member genes\" and \"non-member genes\" needs to be more clearly explained in the Methods and Figure Legends.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In Figure 2 panels D, E and F and Figure 3 panels D, E and F, we think it would be easier to follow the data presentation if only the differentially expressed genes were shown as the fold enrichment in a bar graph instead of the dot plot. The rest of the granular dot plot data can be presented in supplementary material.\\xa0\\n\\n\\n\\xa0\\n\\n\\nDo analyses use the best possible (most unambiguous) available methods, quantified via appropriate statistical comparisons?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0OK\\n\\n\\n\\xa0\\n\\n\\nAre controls or experimental foundations consistent with established findings in the field? A review that raises concerns regarding inconsistency with widely reproduced observations should list at least 2 examples in the literature of such results. To address this question may occasionally require a supplemental figure that, for example, re-graphs multi-axis data from the primary figure using established axes or gating strategies to demonstrate how results in this paper line up with established understandings. It should not be necessary to defend exactly why these may be different from established truths, although doing so may increase the impact of the study.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0OK\\n\\n\\n\\xa0\\n\\n\\n2.\\xa0\\xa0\\xa0\\xa0\\xa0Quality: Completeness (1-3 scale) SCORE =\\xa02\\n\\n\\nDoes the collection of experiments and associated analysis of data support the proposed title/abstract-level conclusions? Typically, the major (title or abstract level) conclusions are expected to be supported by at least two experimental systems.\\xa0\\n\\n\\n\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0It was clear that the bacterial clearance of both isolates was partially due to neutrophil recruitment to the infected lung. It was also clear that the different bacterial loads between the two isolates in lungs at 24h post infection was driven by the RafR SNP. However, the direct evidence to support the conclusion that the IL-17 expression and its downstream neutrophil recruitment in the infected lung contributed to the RafR SNP-mediated phenotypic differences between the two isolates was not convincing. We believe that there are few points in the data presentation and interpretation (as listed below) that need to be further clarified or explained.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0There were two variables (different isolate and different neutrophil treatment) between \"9-47-Ear isolate with anti-mouse Ly6G treatment\" and \"4559-Blood isolate with isotype control treatment\", as well as between \"4559M isolate with anti-mouse Ly6G treatment\" and \"4559-Blood isolate with isotype control treatment\" in Figure 6C. We feel the authors need to address the rationale of comparing the bacterial burden of \"9-47-Ear isolate with anti-mouse Ly6G treatment\", \"4559M isolate with anti-mouse Ly6G treatment\" and \"4559-Blood isolate with isotype control treatment\" (lines 347-348).\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0The authors should clarify the comparison that was described on lines 340-342 since it was not shown in Figure 6C.\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0We would like to see the comparison of the bacterial loads between lungs infected with different isolates under each treatment, especially under anti-mouse Ly6G treatment and anti-IL17A treatment. We would also like to see whether the difference of bacterial loads between isolates changes when neutrophils are depleted or IL-17A is neutralized.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In line 262-266, the authors wrote that\\xa0\"Common differentially expressed genes of this function include genes encoding interleukin 17F (Il17f) and chemokine ligands (Cxcl2, Cxcl3\\xa0and\\xa0Ccl20), with ascending expression level of response to 9-47M, 9-47-Ear and 4559-Blood\".\\xa0The authors need to indicate the corresponding figure. If it was Figure 3C, the comparison group B (9-47-Ear to 4559-Blood) did not show significant difference.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0We think it is necessary to test the neutrophil recruitment in infected and mock-infected lung with/without anti-IL17A treatment.\\xa0\\n\\n\\n\\xa0\\n\\n\\nAre there experiments or analyses that have not been performed, but if \"true\" would disprove the conclusion (sometimes considered a fatal flaw in the study)? In some cases, a reviewer may propose an alternative conclusion/abstract that is clearly defensible with the experiments as presented, and one solution to \\'completeness\\' here should always be to temper an abstract or remove a conclusion and to discuss this alternative in the discussion section.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0OK\\n\\n\\n\\xa0\\n\\n\\n\\xa0\\n\\n\\n\\xa0\\n\\n\\n3.\\xa0\\xa0\\xa0\\xa0\\xa0Quality: Reproducibility (1-3 scale) SCORE =\\xa01.5\\n\\n\\nFigure by Figure, were experiments repeated per a standard of 3x repeats or 5 mice/cohort etc.?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0OK\\n\\n\\nIs there sufficient \"raw data\" presented to assess rigor of the analysis?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Instead of showing sequencing depth as \"times\", the commonly used methodology is to show the raw read counts, the percentage of reads mapped to the host/S. pneumoniae\\xa0genes, and the percentage of alignments. Those data can be organized as a table and included as one the main results for their sequencing data presentation.\\xa0\\n\\n\\nAre methods for experimentation and analysis adequately outlined to permit reproducibility?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In general, more details for each experiments and data analysis are needed. Most importantly, details are needed for cDNA library preparation, DESeq2 and enrichment analysis.\\n\\n\\n\\xa0\\n\\n\\nIf a \\'discovery\\' dataset is used, has a \\'validation\\' cohort been assessed and/or has the issue of false-discovery been addressed?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0OK\\n\\n\\n\\xa0\\n\\n\\n4.\\xa0\\xa0\\xa0\\xa0\\xa0Quality: Scholarship (1-3 scale), generally not the basis for acceptance/rejection: SCORE =\\xa03\\n\\n\\nHas the author cited and discussed the merits of the relevant data that would argue against their conclusion? Has the author cited and/or discussed the important works that are consistent with their conclusion and which a reader should be especially familiar when considering the work?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In the conclusion, the authors stated that\\xa0\"All these variations, and the downstream consequences thereof, are ultimately sensed by host cells, including epithelial and immune cells, resulting in the observed divergence of host response to the various strains, particularly with respect to expression of genes encoding cytokine and chemokine ligands and receptors, as well as those associated with programmed cell death\"\\xa0without any citations. We think the authors need to provide citations to support this statement or exclude this statement from their conclusion.\\xa0\\n\\n\\n\\xa0\\n\\n\\nSpecific (helpful) comments on grammar/diction, paper structure or data presentation (e.g. change a graph style or color scheme) go in this section, but scores in this area not to be significant bases for decisions.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0The authors could write more clearly and concisely to aid reader comprehension.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0It would helpful if the authors include the RafR mutation site when describing the isolates (e.g. \"9-47-Ear RafRD249\" instead of \"9-47-Ear\").\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0The use of an isolate name, such as \"9-47-Ear, 9-47-EarM\", and rafR gene background, such as \"RafRD249G\", was often mixed in one sentence. This made it harder for the readers to understand the data interpretation.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Figure 3E would be better as a supplemental information.\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0There was an incorrect reference to Figure 6B (nasopharynx) on line 348, which should be to Figure 6C (lungs).\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0There was a mixed usage of capital letters and small letters in Figure panels (ie. Figure 2a vs Figure 6A).\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0There were also many gene names and p-values\\xa0were written out in results e.g. lines 182-203, which could be represented in a table.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In line 288, the authors stated that \"In addition to the above, necroptosis, a programmed cell death, is\\xa0almost significantly enriched (p= 0.07)\\xa0in differentially expressed murine genes because of the rafR swap in the blood isolate background\". We think this kind of statements should be avoided.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0The authors stated that\\xa0\"……with the strains expressing the G249 allele triggering a strong pro-inflammatory IL-17 response in the lungs post-infection. This response leads to an influx of neutrophils to the lungs, resulting in the clearance of bacteria. Conversely, expression of the D249\\xa0rafR\\xa0allele results in a more subdued IL-17 host response, allowing for bacterial persistence in the lungs.\"\\xa0at the end of their conclusion. However, we noticed that this statement was contradictory with the previous and current study. The 4559-Blood isolates (G249 allele) was found to be the persistent isolate in lung (Minhas et al., 2019) and induced less neutrophil recruitment in the lung compare to 9-47-Ear isolate (Figure 5).\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Line 341, the authors need to clarify whether it is \"…higher than 9-37-Ear and 4559\" or \"\"…higher than 9-37-Ear and 4559M\".\\xa0\\n\\n\\nMORE SUBJECTIVE CRITERIA (IMPACT):\\n\\n\\n1.\\xa0\\xa0\\xa0\\xa0\\xa0Impact: Novelty/Fundamental and Broad Interest\\xa0(1-4 scale) SCORE =\\xa02\\n\\n\\nA score here should be accompanied by a statement delineating the most interesting/important conceptual finding(s), as they stand right now with the current scope of the paper. A \\'1\\' would be expected to be understood for the importance by a layperson but would also be of top interest (will have lasting impact) on the field.\\xa0\\n\\n\\nHow big of an advance would you consider the findings to be if fully supported but not extended? It would be appropriate to cite literature to provide context for evaluating the advance. However, great care must be taken to avoid exaggerating what is known comparing these findings to the current dogma (see Table 2). Citations (figure by figure) are essential here.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0The authors demonstrated that dual RNA-seq analysis is an excellent approach to study host-pathogen interaction at the transcriptomic level. They identified multiple genes in bacteria and host cells that were affected by RafRD249G. Several molecular mechanisms were proposed to facilitate the interaction between\\xa0S. pneumoniae\\xa0isolates and murine lung. However, the results of the microbiological and immunological tests in this study were not enough to support those predictions. This is a major drawback in the study because it is important to validate the findings from transcriptomic studies with the functional characterization of any host-pathogen interaction. Part of this issue can be solved by improving the presentation and interpretation of existing data.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0We also think that the implications and significance of this study could be expanded by addressing the rationale behind the choice of the two isolates, such as the epidemiology or prevalence of these strains in a clinical setting.\\n\\n\\n\\xa0\\n\\n\\n2.\\xa0\\xa0\\xa0\\xa0\\xa0Impact: Extensibility\\xa0(1-4 scale) SCORE =\\xa0N/A\\n\\n\\nHas an initial result (e.g. of a paradigm in a cell line) been extended to be shown (or implicated) to be important in a bigger scheme (e.g. in animals, or in a human cohort)?\\xa0\\n\\n\\nThis criterion is only valuable as a scoring parameter if it is present, indicated by the N/A option if it simply doesn\\'t apply. The extent to which this is necessary for a result to be considered of value is important. It should be explicitly discussed by a reviewer why it would be required. What work (scope and expected time) and/or discussion would improve this score, and what would this improvement add to the conclusions of the study? Care should be taken to avoid casually suggesting experiments of great cost (e.g. \\'repeat a mouse-based experiment in humans\\') and difficulty that merely confirm, but do not extend. (see Bad Behaviors, Table 2).\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0N/A\\n\\n\\n\\xa0',\n",
              "      'descriptionType': 'Abstract'}],\n",
              "    'geoLocations': [],\n",
              "    'fundingReferences': [],\n",
              "    'url': 'https://zenodo.org/doi/10.5281/zenodo.3702901',\n",
              "    'contentUrl': None,\n",
              "    'metadataVersion': 5,\n",
              "    'schemaVersion': 'http://datacite.org/schema/kernel-4',\n",
              "    'source': 'mds',\n",
              "    'isActive': True,\n",
              "    'state': 'findable',\n",
              "    'reason': None,\n",
              "    'viewCount': 0,\n",
              "    'downloadCount': 0,\n",
              "    'referenceCount': 0,\n",
              "    'citationCount': 0,\n",
              "    'partCount': 0,\n",
              "    'partOfCount': 1,\n",
              "    'versionCount': 0,\n",
              "    'versionOfCount': 1,\n",
              "    'created': '2020-03-10T14:18:02Z',\n",
              "    'registered': '2020-03-10T14:18:04Z',\n",
              "    'published': None,\n",
              "    'updated': '2024-06-03T12:02:13Z'},\n",
              "   'relationships': {'client': {'data': {'id': 'cern.zenodo',\n",
              "      'type': 'clients'}}}},\n",
              "  {'id': '10.5281/zenodo.3702900',\n",
              "   'type': 'dois',\n",
              "   'attributes': {'doi': '10.5281/zenodo.3702900',\n",
              "    'identifiers': [],\n",
              "    'creators': [{'name': 'Craig McCormick',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Craig McCormick',\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0003-2794-3722',\n",
              "        'nameIdentifierScheme': 'ORCID'}],\n",
              "      'affiliation': []}],\n",
              "    'titles': [{'title': 'Review of In Vivo Dual RNA-Seq Analysis Reveals the Basis for Differential Tissue Tropism of Clinical Isolates of Streptococcus pneumoniae'}],\n",
              "    'publisher': 'Zenodo',\n",
              "    'container': {'type': 'Series',\n",
              "     'identifier': 'https://zenodo.org/communities/prereview-reviews',\n",
              "     'identifierType': 'URL'},\n",
              "    'publicationYear': 2020,\n",
              "    'subjects': [],\n",
              "    'contributors': [],\n",
              "    'dates': [{'date': '2020-03-10', 'dateType': 'Issued'}],\n",
              "    'language': 'en',\n",
              "    'types': {'ris': 'GEN',\n",
              "     'bibtex': 'misc',\n",
              "     'citeproc': 'article',\n",
              "     'schemaOrg': 'Review',\n",
              "     'resourceType': '',\n",
              "     'resourceTypeGeneral': 'PeerReview'},\n",
              "    'relatedIdentifiers': [{'relationType': 'Reviews',\n",
              "      'relatedIdentifier': '10.1101/862755',\n",
              "      'resourceTypeGeneral': 'Preprint',\n",
              "      'relatedIdentifierType': 'DOI'},\n",
              "     {'relationType': 'IsIdenticalTo',\n",
              "      'relatedIdentifier': 'https://prereview.org/reviews/3702901',\n",
              "      'resourceTypeGeneral': 'PeerReview',\n",
              "      'relatedIdentifierType': 'URL'},\n",
              "     {'relationType': 'HasVersion',\n",
              "      'relatedIdentifier': '10.5281/zenodo.3702901',\n",
              "      'relatedIdentifierType': 'DOI'}],\n",
              "    'relatedItems': [],\n",
              "    'sizes': [],\n",
              "    'formats': [],\n",
              "    'version': None,\n",
              "    'rightsList': [{'rights': 'Creative Commons Attribution 4.0 International',\n",
              "      'rightsUri': 'https://creativecommons.org/licenses/by/4.0/legalcode',\n",
              "      'schemeUri': 'https://spdx.org/licenses/',\n",
              "      'rightsIdentifier': 'cc-by-4.0',\n",
              "      'rightsIdentifierScheme': 'SPDX'}],\n",
              "    'descriptions': [{'description': 'This Zenodo record is a permanently preserved version of a PREreview. You can view the complete PREreview at https://prereview.org/reviews/3702901.\\n\\n\\nWe, the students of MICI5029/5049, a Graduate Level Molecular Pathogenesis Journal Club at Dalhousie University in Halifax, NS, Canada, hereby submit a review of the following\\xa0BioRxiv\\xa0preprint:\\n\\n\\nVikrant\\xa0Minhas,\\xa0Rieza\\xa0Aprianto,\\xa0Lauren J.\\xa0McAllister,\\xa0Hui\\xa0Wang,\\xa0Shannon C.\\xa0David,\\xa0Kimberley T.\\xa0McLean,\\xa0Iain\\xa0Comerford,\\xa0Shaun R.\\xa0McColl,\\xa0James C.\\xa0Paton,\\xa0Jan-Willem\\xa0Veening,\\xa0Claudia\\xa0Trappetti. In Vivo\\xa0Dual RNA-Seq Analysis Reveals the Basis for Differential Tissue Tropism of Clinical Isolates of\\xa0Streptococcus pneumoniae.\\xa0BioRxiv\\xa0862755; doi:\\xa0https://doi.org/10.1101/862755\\n\\n\\n\\xa0\\n\\n\\nWe will adhere to the Universal Principled (UP) Review guidelines proposed in:\\n\\n\\n\\xa0\\n\\n\\nUniversal Principled Review: A Community-Driven Method to Improve Peer Review.\\xa0Krummel M, Blish C, Kuhns M, Cadwell K, Oberst A, Goldrath A, Ansel KM, Chi H, O\\'Connell R, Wherry EJ, Pepper M; Future Immunology Consortium.\\xa0Cell. 2019 Dec 12;179(7):1441-1445.\\xa0https://doi.org/10.1016/j.cell.2019.11.029\\n\\n\\n\\xa0\\n\\n\\nSUMMARY:\\xa0Previous work has shown that two genetically related clinical isolates (947-Ear RafRD249\\xa0isolate and 4559-Blood RafRG249\\xa0isolate) of\\xa0Streptococcus pneumoniae\\xa0serotype 14 ST15 display differences in virulence in the mouse lung following intranasal challenge. These different virulence phenotypes of the two clinical isolates correlated with non-conserved single nucleotide polymorphisms (SNPs) in the raffinose pathway regulatory gene\\xa0rafR\\xa0(RafRD249G). In this study, Minhas\\xa0et al. investigated the effects of the RafRD249G\\xa0SNP on bacterial and host transcriptomes in infected lungs using a dual RNA-seq approach. They observed differential expression of genes encoding sugar transporters\\xa0that fine-tune carbohydrate metabolism, which were driven by the RafRD249G\\xa0SNP. This contributed to distinct host niche specialization for the two clinical isolates. RNA-seq analysis of the host response to the\\xa0S.\\xa0pneumoniae\\xa0infection\\xa0suggested that the RafRD249G\\xa0SNP in bacteria caused differential expression of host genes encoding for multiple cytokines, cytokine receptors, chemokines and chemokine ligands. Particularly, the expression of IL-17-related genes was enriched in the murine lungs infected with the 947-Ear isolate carrying the RafRD249\\xa0SNP compared to the 947-Ear isolate carrying the RafRG249\\xa0SNP. Using\\xa0in vivo\\xa0neutrophil depletion and IL-17A neutralization, the authors found that IL-17-induced neutrophil recruitment was partially responsible for the more efficient clearance of isolates of 947-Ear RafRD249\\xa0observed in the murine lungs. Overall, this study showed that the RafRD249G\\xa0SNP had a significant impact on bacterial and host transcriptomes, leading to distinct virulence phenotypes and host response. The use of dual RNA-seq in this study provided a powerful approach for investigating\\xa0in vivo\\xa0host-pathogen interactions.\\n\\n\\n\\xa0\\n\\n\\nOVERALL ASSESSMENT:\\xa0\\n\\n\\n\\xa0\\n\\n\\nSTRENGTHS: The authors conducted a transcriptomic study on the impact of a non-conservative SNPs RafRD249G\\xa0on host cells and two\\xa0S. pneumoniae\\xa0isolates (ear and blood) using a dual RNA-seq approach in murine lung infection. Their RNA-seq data strongly suggested that the RafRD249G\\xa0SNP between the two isolates caused distinct patterns of gene expression. Moreover, the dual RNA-seq on the host side showed that the RafRD249G\\xa0SNP also played a detrimental role in trigging differential host responses in infected lung. This study provides solid evidence that the dual RNA-seq approach is a useful tool for\\xa0in vivo\\xa0study of complex host-pathogen interactions.\\xa0\\n\\n\\n\\xa0\\n\\n\\nWEAKNESSES: There are a few weaknesses in the data analysis, data presentation, and experimental design. The lack of information in RNA-seq data analysis made it difficult for the reviewers to evaluate data quality. Some of the figure labels are difficult to interpret. In certain instances, methods were not described in sufficient detail to enable others to reproduce experiments. In this study, the RNA-seq data suggested an association between IL-17 expression and the RafRD249G\\xa0SNP in bacteria. However, the conclusion that RafRD249G\\xa0SNP was responsible for bacterial clearance via IL-17-mediated neutrophil recruitment was insufficiently supported. This could be improved by using appropriate controls, testing neutrophil recruitment in murine lung, and comparing data within each treatment. Moreover, we found that some of the writing in the manuscript was misleading and confusing. We believe that some of the weaknesses mentioned above could be solved by simplifying and clarifying the data presentation and interpretation.\\n\\n\\n\\xa0\\n\\n\\nDETAILED U.P. ASSESSMENT:\\n\\n\\n\\xa0\\n\\n\\nOBJECTIVE CRITERIA (QUALITY):\\xa0\\n\\n\\n\\xa0\\n\\n\\n1.\\xa0\\xa0\\xa0\\xa0\\xa0Quality: Experiments (1-3 scale) SCORE =\\xa01.5\\n\\n\\nFigure by Figure, do experiments, as performed, have the proper controls?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Figures 2B and 2C: There appears to be an enrichment in higher fold changes on either side of the replication origin. Have the authors considered whether cell populations might be rapidly growing, which could provide multiple copies of the origin per cell, thereby clouding interpretation?\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Figure 5: The gating strategy for multi-color flow cytometry analysis should be provided.\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Figure 5 and 6A: We think adding a mock infected negative control or showing the data of time zero as the base line is necessary to interpret this data.\\xa0\\n\\n\\n\\xa0\\n\\n\\nAre specific analyses performed using methods that are consistent with answering the specific question? Is there the appropriate technical expertise in the collection and analysis of data presented?\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In Figure 2&3, the meaning of \"member genes\" and \"non-member genes\" needs to be more clearly explained in the Methods and Figure Legends.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In Figure 2 panels D, E and F and Figure 3 panels D, E and F, we think it would be easier to follow the data presentation if only the differentially expressed genes were shown as the fold enrichment in a bar graph instead of the dot plot. The rest of the granular dot plot data can be presented in supplementary material.\\xa0\\n\\n\\n\\xa0\\n\\n\\nDo analyses use the best possible (most unambiguous) available methods, quantified via appropriate statistical comparisons?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0OK\\n\\n\\n\\xa0\\n\\n\\nAre controls or experimental foundations consistent with established findings in the field? A review that raises concerns regarding inconsistency with widely reproduced observations should list at least 2 examples in the literature of such results. To address this question may occasionally require a supplemental figure that, for example, re-graphs multi-axis data from the primary figure using established axes or gating strategies to demonstrate how results in this paper line up with established understandings. It should not be necessary to defend exactly why these may be different from established truths, although doing so may increase the impact of the study.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0OK\\n\\n\\n\\xa0\\n\\n\\n2.\\xa0\\xa0\\xa0\\xa0\\xa0Quality: Completeness (1-3 scale) SCORE =\\xa02\\n\\n\\nDoes the collection of experiments and associated analysis of data support the proposed title/abstract-level conclusions? Typically, the major (title or abstract level) conclusions are expected to be supported by at least two experimental systems.\\xa0\\n\\n\\n\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0It was clear that the bacterial clearance of both isolates was partially due to neutrophil recruitment to the infected lung. It was also clear that the different bacterial loads between the two isolates in lungs at 24h post infection was driven by the RafR SNP. However, the direct evidence to support the conclusion that the IL-17 expression and its downstream neutrophil recruitment in the infected lung contributed to the RafR SNP-mediated phenotypic differences between the two isolates was not convincing. We believe that there are few points in the data presentation and interpretation (as listed below) that need to be further clarified or explained.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0There were two variables (different isolate and different neutrophil treatment) between \"9-47-Ear isolate with anti-mouse Ly6G treatment\" and \"4559-Blood isolate with isotype control treatment\", as well as between \"4559M isolate with anti-mouse Ly6G treatment\" and \"4559-Blood isolate with isotype control treatment\" in Figure 6C. We feel the authors need to address the rationale of comparing the bacterial burden of \"9-47-Ear isolate with anti-mouse Ly6G treatment\", \"4559M isolate with anti-mouse Ly6G treatment\" and \"4559-Blood isolate with isotype control treatment\" (lines 347-348).\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0The authors should clarify the comparison that was described on lines 340-342 since it was not shown in Figure 6C.\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0We would like to see the comparison of the bacterial loads between lungs infected with different isolates under each treatment, especially under anti-mouse Ly6G treatment and anti-IL17A treatment. We would also like to see whether the difference of bacterial loads between isolates changes when neutrophils are depleted or IL-17A is neutralized.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In line 262-266, the authors wrote that\\xa0\"Common differentially expressed genes of this function include genes encoding interleukin 17F (Il17f) and chemokine ligands (Cxcl2, Cxcl3\\xa0and\\xa0Ccl20), with ascending expression level of response to 9-47M, 9-47-Ear and 4559-Blood\".\\xa0The authors need to indicate the corresponding figure. If it was Figure 3C, the comparison group B (9-47-Ear to 4559-Blood) did not show significant difference.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0We think it is necessary to test the neutrophil recruitment in infected and mock-infected lung with/without anti-IL17A treatment.\\xa0\\n\\n\\n\\xa0\\n\\n\\nAre there experiments or analyses that have not been performed, but if \"true\" would disprove the conclusion (sometimes considered a fatal flaw in the study)? In some cases, a reviewer may propose an alternative conclusion/abstract that is clearly defensible with the experiments as presented, and one solution to \\'completeness\\' here should always be to temper an abstract or remove a conclusion and to discuss this alternative in the discussion section.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0OK\\n\\n\\n\\xa0\\n\\n\\n\\xa0\\n\\n\\n\\xa0\\n\\n\\n3.\\xa0\\xa0\\xa0\\xa0\\xa0Quality: Reproducibility (1-3 scale) SCORE =\\xa01.5\\n\\n\\nFigure by Figure, were experiments repeated per a standard of 3x repeats or 5 mice/cohort etc.?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0OK\\n\\n\\nIs there sufficient \"raw data\" presented to assess rigor of the analysis?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Instead of showing sequencing depth as \"times\", the commonly used methodology is to show the raw read counts, the percentage of reads mapped to the host/S. pneumoniae\\xa0genes, and the percentage of alignments. Those data can be organized as a table and included as one the main results for their sequencing data presentation.\\xa0\\n\\n\\nAre methods for experimentation and analysis adequately outlined to permit reproducibility?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In general, more details for each experiments and data analysis are needed. Most importantly, details are needed for cDNA library preparation, DESeq2 and enrichment analysis.\\n\\n\\n\\xa0\\n\\n\\nIf a \\'discovery\\' dataset is used, has a \\'validation\\' cohort been assessed and/or has the issue of false-discovery been addressed?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0OK\\n\\n\\n\\xa0\\n\\n\\n4.\\xa0\\xa0\\xa0\\xa0\\xa0Quality: Scholarship (1-3 scale), generally not the basis for acceptance/rejection: SCORE =\\xa03\\n\\n\\nHas the author cited and discussed the merits of the relevant data that would argue against their conclusion? Has the author cited and/or discussed the important works that are consistent with their conclusion and which a reader should be especially familiar when considering the work?\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In the conclusion, the authors stated that\\xa0\"All these variations, and the downstream consequences thereof, are ultimately sensed by host cells, including epithelial and immune cells, resulting in the observed divergence of host response to the various strains, particularly with respect to expression of genes encoding cytokine and chemokine ligands and receptors, as well as those associated with programmed cell death\"\\xa0without any citations. We think the authors need to provide citations to support this statement or exclude this statement from their conclusion.\\xa0\\n\\n\\n\\xa0\\n\\n\\nSpecific (helpful) comments on grammar/diction, paper structure or data presentation (e.g. change a graph style or color scheme) go in this section, but scores in this area not to be significant bases for decisions.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0The authors could write more clearly and concisely to aid reader comprehension.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0It would helpful if the authors include the RafR mutation site when describing the isolates (e.g. \"9-47-Ear RafRD249\" instead of \"9-47-Ear\").\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0The use of an isolate name, such as \"9-47-Ear, 9-47-EarM\", and rafR gene background, such as \"RafRD249G\", was often mixed in one sentence. This made it harder for the readers to understand the data interpretation.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Figure 3E would be better as a supplemental information.\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0There was an incorrect reference to Figure 6B (nasopharynx) on line 348, which should be to Figure 6C (lungs).\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0There was a mixed usage of capital letters and small letters in Figure panels (ie. Figure 2a vs Figure 6A).\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0There were also many gene names and p-values\\xa0were written out in results e.g. lines 182-203, which could be represented in a table.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0In line 288, the authors stated that \"In addition to the above, necroptosis, a programmed cell death, is\\xa0almost significantly enriched (p= 0.07)\\xa0in differentially expressed murine genes because of the rafR swap in the blood isolate background\". We think this kind of statements should be avoided.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0The authors stated that\\xa0\"……with the strains expressing the G249 allele triggering a strong pro-inflammatory IL-17 response in the lungs post-infection. This response leads to an influx of neutrophils to the lungs, resulting in the clearance of bacteria. Conversely, expression of the D249\\xa0rafR\\xa0allele results in a more subdued IL-17 host response, allowing for bacterial persistence in the lungs.\"\\xa0at the end of their conclusion. However, we noticed that this statement was contradictory with the previous and current study. The 4559-Blood isolates (G249 allele) was found to be the persistent isolate in lung (Minhas et al., 2019) and induced less neutrophil recruitment in the lung compare to 9-47-Ear isolate (Figure 5).\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0Line 341, the authors need to clarify whether it is \"…higher than 9-37-Ear and 4559\" or \"\"…higher than 9-37-Ear and 4559M\".\\xa0\\n\\n\\nMORE SUBJECTIVE CRITERIA (IMPACT):\\n\\n\\n1.\\xa0\\xa0\\xa0\\xa0\\xa0Impact: Novelty/Fundamental and Broad Interest\\xa0(1-4 scale) SCORE =\\xa02\\n\\n\\nA score here should be accompanied by a statement delineating the most interesting/important conceptual finding(s), as they stand right now with the current scope of the paper. A \\'1\\' would be expected to be understood for the importance by a layperson but would also be of top interest (will have lasting impact) on the field.\\xa0\\n\\n\\nHow big of an advance would you consider the findings to be if fully supported but not extended? It would be appropriate to cite literature to provide context for evaluating the advance. However, great care must be taken to avoid exaggerating what is known comparing these findings to the current dogma (see Table 2). Citations (figure by figure) are essential here.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0The authors demonstrated that dual RNA-seq analysis is an excellent approach to study host-pathogen interaction at the transcriptomic level. They identified multiple genes in bacteria and host cells that were affected by RafRD249G. Several molecular mechanisms were proposed to facilitate the interaction between\\xa0S. pneumoniae\\xa0isolates and murine lung. However, the results of the microbiological and immunological tests in this study were not enough to support those predictions. This is a major drawback in the study because it is important to validate the findings from transcriptomic studies with the functional characterization of any host-pathogen interaction. Part of this issue can be solved by improving the presentation and interpretation of existing data.\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0We also think that the implications and significance of this study could be expanded by addressing the rationale behind the choice of the two isolates, such as the epidemiology or prevalence of these strains in a clinical setting.\\n\\n\\n\\xa0\\n\\n\\n2.\\xa0\\xa0\\xa0\\xa0\\xa0Impact: Extensibility\\xa0(1-4 scale) SCORE =\\xa0N/A\\n\\n\\nHas an initial result (e.g. of a paradigm in a cell line) been extended to be shown (or implicated) to be important in a bigger scheme (e.g. in animals, or in a human cohort)?\\xa0\\n\\n\\nThis criterion is only valuable as a scoring parameter if it is present, indicated by the N/A option if it simply doesn\\'t apply. The extent to which this is necessary for a result to be considered of value is important. It should be explicitly discussed by a reviewer why it would be required. What work (scope and expected time) and/or discussion would improve this score, and what would this improvement add to the conclusions of the study? Care should be taken to avoid casually suggesting experiments of great cost (e.g. \\'repeat a mouse-based experiment in humans\\') and difficulty that merely confirm, but do not extend. (see Bad Behaviors, Table 2).\\xa0\\n\\n\\n·\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0\\xa0N/A\\n\\n\\n\\xa0',\n",
              "      'descriptionType': 'Abstract'}],\n",
              "    'geoLocations': [],\n",
              "    'fundingReferences': [],\n",
              "    'url': 'https://zenodo.org/doi/10.5281/zenodo.3702900',\n",
              "    'contentUrl': None,\n",
              "    'metadataVersion': 5,\n",
              "    'schemaVersion': 'http://datacite.org/schema/kernel-4',\n",
              "    'source': 'mds',\n",
              "    'isActive': True,\n",
              "    'state': 'findable',\n",
              "    'reason': None,\n",
              "    'viewCount': 0,\n",
              "    'downloadCount': 0,\n",
              "    'referenceCount': 0,\n",
              "    'citationCount': 0,\n",
              "    'partCount': 0,\n",
              "    'partOfCount': 1,\n",
              "    'versionCount': 1,\n",
              "    'versionOfCount': 0,\n",
              "    'created': '2020-03-10T14:18:04Z',\n",
              "    'registered': '2020-03-10T14:18:05Z',\n",
              "    'published': None,\n",
              "    'updated': '2024-06-03T12:02:13Z'},\n",
              "   'relationships': {'client': {'data': {'id': 'cern.zenodo',\n",
              "      'type': 'clients'}}}},\n",
              "  {'id': '10.60692/efymp-hpb34',\n",
              "   'type': 'dois',\n",
              "   'attributes': {'doi': '10.60692/efymp-hpb34',\n",
              "    'identifiers': [{'identifier': 'oai:gresis.org:efymp-hpb34',\n",
              "      'identifierType': 'oai'},\n",
              "     {'identifier': 'https://openalex.org/W2442839603',\n",
              "      'identifierType': 'Other'}],\n",
              "    'creators': [{'name': 'Parul Gupta',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Parul Gupta',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-0190-8753',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Ridhi Goel',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Ridhi Goel',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Aishwarya Agarwal',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Aishwarya Agarwal',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-5368-2442',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Mehar Hasan Asif',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Mehar Hasan Asif',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-7640-4779',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Neelam S. Sangwan',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Neelam S. Sangwan',\n",
              "      'affiliation': ['Central Institute of Medicinal and Aromatic Plants'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Neelam S. Sangwan',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Neelam S. Sangwan',\n",
              "      'affiliation': ['Central Institute of Medicinal and Aromatic Plants'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-7766-5428',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Prabodh Kumar Trivedi',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Prabodh Kumar Trivedi',\n",
              "      'affiliation': ['Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0001-6463-1731',\n",
              "        'nameIdentifierScheme': 'ORCID'}]}],\n",
              "    'titles': [{'title': 'Comparative transcriptome analysis of different chemotypes elucidates withanolide biosynthesis pathway from medicinal plant Withania somnifera'},\n",
              "     {'lang': 'ara',\n",
              "      'title': 'تحليل النسخ المقارن للأنماط الكيميائية المختلفة يوضح مسار التخليق الحيوي للـ withanolide من النبات الطبي Withania somnifera',\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'fra',\n",
              "      'title': \"L'analyse comparative du transcriptome de différents chimiotypes élucide la voie de biosynthèse du withanolide à partir de la plante médicinale Withania somnifera\",\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'spa',\n",
              "      'title': 'El análisis comparativo del transcriptoma de diferentes quimiotipos aclara la vía de biosíntesis de withanólidos de la planta medicinal Withania somnifera',\n",
              "      'titleType': 'TranslatedTitle'}],\n",
              "    'publisher': 'OpenAlex',\n",
              "    'container': {},\n",
              "    'publicationYear': 2015,\n",
              "    'subjects': [{'subject': 'Therapeutic Potential of Ayurvedic Medicine'},\n",
              "     {'subject': 'Complementary and alternative medicine'},\n",
              "     {'subject': 'Medicine'},\n",
              "     {'subject': 'Health Sciences'},\n",
              "     {'subject': 'Herbal Medicine for Neurological Disorders'},\n",
              "     {'subject': 'Withania somnifera'},\n",
              "     {'subject': 'Withanolide'},\n",
              "     {'subject': 'Chemotype'},\n",
              "     {'subject': 'Transcriptome'},\n",
              "     {'subject': 'Biology'},\n",
              "     {'subject': 'Botany'},\n",
              "     {'subject': 'Computational biology'},\n",
              "     {'subject': 'Traditional medicine'},\n",
              "     {'subject': 'Genetics'},\n",
              "     {'subject': 'FOS: Biological sciences',\n",
              "      'schemeUri': 'http://www.oecd.org/science/inno/38235147.pdf',\n",
              "      'subjectScheme': 'Fields of Science and Technology (FOS)'},\n",
              "     {'subject': 'Gene'},\n",
              "     {'subject': 'Gene expression'},\n",
              "     {'subject': 'Alternative medicine'},\n",
              "     {'subject': 'Pathology'},\n",
              "     {'subject': 'Essential oil'}],\n",
              "    'contributors': [],\n",
              "    'dates': [{'date': '2015-12-21', 'dateType': 'Issued'}],\n",
              "    'language': 'en',\n",
              "    'types': {'ris': 'RPRT',\n",
              "     'bibtex': 'article',\n",
              "     'citeproc': 'article-journal',\n",
              "     'schemaOrg': 'ScholarlyArticle',\n",
              "     'resourceType': '',\n",
              "     'resourceTypeGeneral': 'Text'},\n",
              "    'relatedIdentifiers': [{'relationType': 'IsVersionOf',\n",
              "      'relatedIdentifier': '10.60692/4jv94-r2r41',\n",
              "      'relatedIdentifierType': 'DOI'}],\n",
              "    'relatedItems': [],\n",
              "    'sizes': [],\n",
              "    'formats': [],\n",
              "    'version': 'v1',\n",
              "    'rightsList': [{'rights': 'cc-by'}],\n",
              "    'descriptions': [{'description': 'Withania somnifera is one of the most valuable medicinal plants synthesizing secondary metabolites known as withanolides. Despite pharmaceutical importance, limited information is available about the biosynthesis of withanolides. Chemo-profiling of leaf and root tissues of Withania suggest differences in the content and/or nature of withanolides in different chemotypes. To identify genes involved in chemotype and/or tissue-specific withanolide biosynthesis, we established transcriptomes of leaf and root tissues of distinct chemotypes. Genes encoding enzymes for intermediate steps of terpenoid backbone biosynthesis with their alternatively spliced forms and paralogous have been identified. Analysis suggests differential expression of large number genes among leaf and root tissues of different chemotypes. Study also identified differentially expressing transcripts encoding cytochrome P450s, glycosyltransferases, methyltransferases and transcription factors which might be involved in chemodiversity in Withania. Virus induced gene silencing of the sterol ∆7-reductase (WsDWF5) involved in the synthesis of 24-methylene cholesterol, withanolide backbone, suggests role of this enzyme in biosynthesis of withanolides. Information generated, in this study, provides a rich resource for functional analysis of withanolide-specific genes to elucidate chemotype- as well as tissue-specific withanolide biosynthesis. This genomic resource will also help in development of new tools for functional genomics and breeding in Withania.',\n",
              "      'descriptionType': 'Abstract'},\n",
              "     {'lang': 'ara',\n",
              "      'description': 'ويثانيا سومنيفيرا هي واحدة من النباتات الطبية الأكثر قيمة التي تقوم بتوليف المستقلبات الثانوية المعروفة باسم ويثانوليدس. على الرغم من الأهمية الصيدلانية، تتوفر معلومات محدودة حول التركيب الحيوي للويثانوليدات. يشير التنميط الكيميائي لأنسجة الأوراق والجذور في ويثانيا إلى اختلافات في محتوى و/أو طبيعة ويثانوليدس في أنماط كيميائية مختلفة. لتحديد الجينات المشاركة في النمط الكيميائي و/أو التخليق الحيوي للأنسجة الخاصة بالـ withanolide، أنشأنا transcriptomes من أنسجة الأوراق والجذور من الأنماط الكيميائية المتميزة. تم تحديد إنزيمات تشفير الجينات للخطوات الوسيطة للتخليق الحيوي للعمود الفقري التربيناني مع أشكالها المجزأة البديلة والمتماثلة. يشير التحليل إلى التعبير التفاضلي عن عدد كبير من الجينات بين أنسجة الأوراق والجذور من أنماط كيميائية مختلفة. كما حددت الدراسة بشكل تفاضلي النصوص المعبرة التي تشفر السيتوكروم P450s، وناقلات الجليكوسيل، وناقلات الميثيل وعوامل النسخ التي قد تشارك في التنوع الكيميائي في ويثانيا. يشير إسكات الجينات المستحثة بالفيروس من ستيرول 7 -ريدوكتاز (WsDWF5) المشاركة في تخليق 24 -ميثيلين الكوليسترول، العمود الفقري للويثانوليد، إلى دور هذا الإنزيم في التخليق الحيوي للويثانوليدات. توفر المعلومات التي تم إنشاؤها، في هذه الدراسة، موردًا غنيًا للتحليل الوظيفي للجينات الخاصة بالويثانوليد لتوضيح النمط الكيميائي - وكذلك التخليق الحيوي للويثانوليد الخاص بالأنسجة. سيساعد هذا المورد الجيني أيضًا في تطوير أدوات جديدة لعلم الجينوم الوظيفي والتربية في ويثانيا.',\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'fra',\n",
              "      'description': \"Withania somnifera est l'une des plantes médicinales les plus précieuses synthétisant des métabolites secondaires connus sous le nom de withanolides. Malgré l'importance pharmaceutique, peu d'informations sont disponibles sur la biosynthèse des withanolides. Le profilage chimio des tissus foliaires et racinaires de Withania suggère des différences dans la teneur et/ou la nature des withanolides dans différents chimiotypes. Pour identifier les gènes impliqués dans la biosynthèse du chémotype et/ou du tissu spécifique avec l'éthanolide, nous avons établi les transcriptomes des tissus foliaires et racinaires de chémotypes distincts. Des gènes codant des enzymes pour des étapes intermédiaires de la biosynthèse du squelette terpénoïde avec leurs formes alternativement épissées et paralogues ont été identifiés. L'analyse suggère l'expression différentielle d'un grand nombre de gènes parmi les tissus foliaires et racinaires de différents chémotypes. L'étude a également identifié des transcrits à expression différentielle codant pour les cytochromes P450, les glycosyltransférases, les méthyltransférases et les facteurs de transcription qui pourraient être impliqués dans la chimiodiversité chez Withania. Le silençage génique induit par le virus de la stérol ∆7-réductase (WsDWF5) impliquée dans la synthèse du 24-méthylène cholestérol, squelette du withanolide, suggère le rôle de cette enzyme dans la biosynthèse des withanolides. Les informations générées dans cette étude constituent une riche ressource pour l'analyse fonctionnelle des gènes spécifiques du withanolide afin d'élucider la biosynthèse du withanolide chimiotypique et tissulaire. Cette ressource génomique contribuera également au développement de nouveaux outils pour la génomique fonctionnelle et la sélection à Withania.\",\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'spa',\n",
              "      'description': 'Withania somnifera es una de las plantas medicinales más valiosas que sintetizan metabolitos secundarios conocidos como withanólidos. A pesar de la importancia farmacéutica, se dispone de información limitada sobre la biosíntesis de los withanólidos. El perfil químico de los tejidos foliares y radiculares de Withania sugiere diferencias en el contenido y/o la naturaleza de los withanólidos en diferentes quimiotipos. Para identificar los genes implicados en el quimiotipo y/o la biosíntesis de withanolida específica de tejido, establecimos transcriptomas de tejidos de hojas y raíces de distintos quimiotipos. Se han identificado genes que codifican enzimas para etapas intermedias de la biosíntesis de la estructura principal terpenoide con sus formas alternativamente empalmadas y parálogas. El análisis sugiere una expresión diferencial de un gran número de genes entre los tejidos de hojas y raíces de diferentes quimiotipos. El estudio también identificó transcritos de expresión diferencial que codifican citocromo P450, glicosiltransferasas, metiltransferasas y factores de transcripción que podrían estar involucrados en la quimiodiversidad en Withania. El silenciamiento génico inducido por virus de la esterol ∆7-reductasa (WsDWF5) involucrada en la síntesis de colesterol 24-metileno, estructura principal de withanolida, sugiere el papel de esta enzima en la biosíntesis de withanolidas. La información generada, en este estudio, proporciona un recurso rico para el análisis funcional de genes específicos de withanolida para dilucidar la biosíntesis de withanolida específica del quimiotipo y del tejido. Este recurso genómico también ayudará en el desarrollo de nuevas herramientas para la genómica funcional y la cría en Withania.',\n",
              "      'descriptionType': 'Other'}],\n",
              "    'geoLocations': [],\n",
              "    'fundingReferences': [],\n",
              "    'url': 'https://gresis.osc.int//doi/10.60692/efymp-hpb34',\n",
              "    'contentUrl': None,\n",
              "    'metadataVersion': 0,\n",
              "    'schemaVersion': 'http://datacite.org/schema/kernel-4',\n",
              "    'source': 'api',\n",
              "    'isActive': True,\n",
              "    'state': 'findable',\n",
              "    'reason': None,\n",
              "    'viewCount': 0,\n",
              "    'downloadCount': 0,\n",
              "    'referenceCount': 0,\n",
              "    'citationCount': 0,\n",
              "    'partCount': 0,\n",
              "    'partOfCount': 0,\n",
              "    'versionCount': 0,\n",
              "    'versionOfCount': 0,\n",
              "    'created': '2024-06-03T10:55:10Z',\n",
              "    'registered': '2024-06-03T10:55:11Z',\n",
              "    'published': None,\n",
              "    'updated': '2024-06-03T10:55:11Z'},\n",
              "   'relationships': {'client': {'data': {'id': 'iapx.bsycxq',\n",
              "      'type': 'clients'}}}},\n",
              "  {'id': '10.60692/4jv94-r2r41',\n",
              "   'type': 'dois',\n",
              "   'attributes': {'doi': '10.60692/4jv94-r2r41',\n",
              "    'identifiers': [{'identifier': 'https://openalex.org/W2442839603',\n",
              "      'identifierType': 'Other'}],\n",
              "    'creators': [{'name': 'Parul Gupta',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Parul Gupta',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-0190-8753',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Ridhi Goel',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Ridhi Goel',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Aishwarya Agarwal',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Aishwarya Agarwal',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-5368-2442',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Mehar Hasan Asif',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Mehar Hasan Asif',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-7640-4779',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Neelam S. Sangwan',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Neelam S. Sangwan',\n",
              "      'affiliation': ['Central Institute of Medicinal and Aromatic Plants'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Neelam S. Sangwan',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Neelam S. Sangwan',\n",
              "      'affiliation': ['Central Institute of Medicinal and Aromatic Plants'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-7766-5428',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Prabodh Kumar Trivedi',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Prabodh Kumar Trivedi',\n",
              "      'affiliation': ['Academy of Scientific and Innovative Research'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0001-6463-1731',\n",
              "        'nameIdentifierScheme': 'ORCID'}]}],\n",
              "    'titles': [{'title': 'Comparative transcriptome analysis of different chemotypes elucidates withanolide biosynthesis pathway from medicinal plant Withania somnifera'},\n",
              "     {'lang': 'ara',\n",
              "      'title': 'تحليل النسخ المقارن للأنماط الكيميائية المختلفة يوضح مسار التخليق الحيوي للـ withanolide من النبات الطبي Withania somnifera',\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'fra',\n",
              "      'title': \"L'analyse comparative du transcriptome de différents chimiotypes élucide la voie de biosynthèse du withanolide à partir de la plante médicinale Withania somnifera\",\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'spa',\n",
              "      'title': 'El análisis comparativo del transcriptoma de diferentes quimiotipos aclara la vía de biosíntesis de withanólidos de la planta medicinal Withania somnifera',\n",
              "      'titleType': 'TranslatedTitle'}],\n",
              "    'publisher': 'OpenAlex',\n",
              "    'container': {},\n",
              "    'publicationYear': 2015,\n",
              "    'subjects': [{'subject': 'Therapeutic Potential of Ayurvedic Medicine'},\n",
              "     {'subject': 'Complementary and alternative medicine'},\n",
              "     {'subject': 'Medicine'},\n",
              "     {'subject': 'Health Sciences'},\n",
              "     {'subject': 'Herbal Medicine for Neurological Disorders'},\n",
              "     {'subject': 'Withania somnifera'},\n",
              "     {'subject': 'Withanolide'},\n",
              "     {'subject': 'Chemotype'},\n",
              "     {'subject': 'Transcriptome'},\n",
              "     {'subject': 'Biology'},\n",
              "     {'subject': 'Botany'},\n",
              "     {'subject': 'Computational biology'},\n",
              "     {'subject': 'Traditional medicine'},\n",
              "     {'subject': 'Genetics'},\n",
              "     {'subject': 'FOS: Biological sciences',\n",
              "      'schemeUri': 'http://www.oecd.org/science/inno/38235147.pdf',\n",
              "      'subjectScheme': 'Fields of Science and Technology (FOS)'},\n",
              "     {'subject': 'Gene'},\n",
              "     {'subject': 'Gene expression'},\n",
              "     {'subject': 'Alternative medicine'},\n",
              "     {'subject': 'Pathology'},\n",
              "     {'subject': 'Essential oil'}],\n",
              "    'contributors': [],\n",
              "    'dates': [{'date': '2015-12-21', 'dateType': 'Issued'}],\n",
              "    'language': 'en',\n",
              "    'types': {'ris': 'RPRT',\n",
              "     'bibtex': 'article',\n",
              "     'citeproc': 'article-journal',\n",
              "     'schemaOrg': 'ScholarlyArticle',\n",
              "     'resourceType': '',\n",
              "     'resourceTypeGeneral': 'Text'},\n",
              "    'relatedIdentifiers': [{'relationType': 'HasVersion',\n",
              "      'relatedIdentifier': '10.60692/efymp-hpb34',\n",
              "      'relatedIdentifierType': 'DOI'}],\n",
              "    'relatedItems': [],\n",
              "    'sizes': [],\n",
              "    'formats': [],\n",
              "    'version': 'v1',\n",
              "    'rightsList': [{'rights': 'cc-by'}],\n",
              "    'descriptions': [{'description': 'Withania somnifera is one of the most valuable medicinal plants synthesizing secondary metabolites known as withanolides. Despite pharmaceutical importance, limited information is available about the biosynthesis of withanolides. Chemo-profiling of leaf and root tissues of Withania suggest differences in the content and/or nature of withanolides in different chemotypes. To identify genes involved in chemotype and/or tissue-specific withanolide biosynthesis, we established transcriptomes of leaf and root tissues of distinct chemotypes. Genes encoding enzymes for intermediate steps of terpenoid backbone biosynthesis with their alternatively spliced forms and paralogous have been identified. Analysis suggests differential expression of large number genes among leaf and root tissues of different chemotypes. Study also identified differentially expressing transcripts encoding cytochrome P450s, glycosyltransferases, methyltransferases and transcription factors which might be involved in chemodiversity in Withania. Virus induced gene silencing of the sterol ∆7-reductase (WsDWF5) involved in the synthesis of 24-methylene cholesterol, withanolide backbone, suggests role of this enzyme in biosynthesis of withanolides. Information generated, in this study, provides a rich resource for functional analysis of withanolide-specific genes to elucidate chemotype- as well as tissue-specific withanolide biosynthesis. This genomic resource will also help in development of new tools for functional genomics and breeding in Withania.',\n",
              "      'descriptionType': 'Abstract'},\n",
              "     {'lang': 'ara',\n",
              "      'description': 'ويثانيا سومنيفيرا هي واحدة من النباتات الطبية الأكثر قيمة التي تقوم بتوليف المستقلبات الثانوية المعروفة باسم ويثانوليدس. على الرغم من الأهمية الصيدلانية، تتوفر معلومات محدودة حول التركيب الحيوي للويثانوليدات. يشير التنميط الكيميائي لأنسجة الأوراق والجذور في ويثانيا إلى اختلافات في محتوى و/أو طبيعة ويثانوليدس في أنماط كيميائية مختلفة. لتحديد الجينات المشاركة في النمط الكيميائي و/أو التخليق الحيوي للأنسجة الخاصة بالـ withanolide، أنشأنا transcriptomes من أنسجة الأوراق والجذور من الأنماط الكيميائية المتميزة. تم تحديد إنزيمات تشفير الجينات للخطوات الوسيطة للتخليق الحيوي للعمود الفقري التربيناني مع أشكالها المجزأة البديلة والمتماثلة. يشير التحليل إلى التعبير التفاضلي عن عدد كبير من الجينات بين أنسجة الأوراق والجذور من أنماط كيميائية مختلفة. كما حددت الدراسة بشكل تفاضلي النصوص المعبرة التي تشفر السيتوكروم P450s، وناقلات الجليكوسيل، وناقلات الميثيل وعوامل النسخ التي قد تشارك في التنوع الكيميائي في ويثانيا. يشير إسكات الجينات المستحثة بالفيروس من ستيرول 7 -ريدوكتاز (WsDWF5) المشاركة في تخليق 24 -ميثيلين الكوليسترول، العمود الفقري للويثانوليد، إلى دور هذا الإنزيم في التخليق الحيوي للويثانوليدات. توفر المعلومات التي تم إنشاؤها، في هذه الدراسة، موردًا غنيًا للتحليل الوظيفي للجينات الخاصة بالويثانوليد لتوضيح النمط الكيميائي - وكذلك التخليق الحيوي للويثانوليد الخاص بالأنسجة. سيساعد هذا المورد الجيني أيضًا في تطوير أدوات جديدة لعلم الجينوم الوظيفي والتربية في ويثانيا.',\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'fra',\n",
              "      'description': \"Withania somnifera est l'une des plantes médicinales les plus précieuses synthétisant des métabolites secondaires connus sous le nom de withanolides. Malgré l'importance pharmaceutique, peu d'informations sont disponibles sur la biosynthèse des withanolides. Le profilage chimio des tissus foliaires et racinaires de Withania suggère des différences dans la teneur et/ou la nature des withanolides dans différents chimiotypes. Pour identifier les gènes impliqués dans la biosynthèse du chémotype et/ou du tissu spécifique avec l'éthanolide, nous avons établi les transcriptomes des tissus foliaires et racinaires de chémotypes distincts. Des gènes codant des enzymes pour des étapes intermédiaires de la biosynthèse du squelette terpénoïde avec leurs formes alternativement épissées et paralogues ont été identifiés. L'analyse suggère l'expression différentielle d'un grand nombre de gènes parmi les tissus foliaires et racinaires de différents chémotypes. L'étude a également identifié des transcrits à expression différentielle codant pour les cytochromes P450, les glycosyltransférases, les méthyltransférases et les facteurs de transcription qui pourraient être impliqués dans la chimiodiversité chez Withania. Le silençage génique induit par le virus de la stérol ∆7-réductase (WsDWF5) impliquée dans la synthèse du 24-méthylène cholestérol, squelette du withanolide, suggère le rôle de cette enzyme dans la biosynthèse des withanolides. Les informations générées dans cette étude constituent une riche ressource pour l'analyse fonctionnelle des gènes spécifiques du withanolide afin d'élucider la biosynthèse du withanolide chimiotypique et tissulaire. Cette ressource génomique contribuera également au développement de nouveaux outils pour la génomique fonctionnelle et la sélection à Withania.\",\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'spa',\n",
              "      'description': 'Withania somnifera es una de las plantas medicinales más valiosas que sintetizan metabolitos secundarios conocidos como withanólidos. A pesar de la importancia farmacéutica, se dispone de información limitada sobre la biosíntesis de los withanólidos. El perfil químico de los tejidos foliares y radiculares de Withania sugiere diferencias en el contenido y/o la naturaleza de los withanólidos en diferentes quimiotipos. Para identificar los genes implicados en el quimiotipo y/o la biosíntesis de withanolida específica de tejido, establecimos transcriptomas de tejidos de hojas y raíces de distintos quimiotipos. Se han identificado genes que codifican enzimas para etapas intermedias de la biosíntesis de la estructura principal terpenoide con sus formas alternativamente empalmadas y parálogas. El análisis sugiere una expresión diferencial de un gran número de genes entre los tejidos de hojas y raíces de diferentes quimiotipos. El estudio también identificó transcritos de expresión diferencial que codifican citocromo P450, glicosiltransferasas, metiltransferasas y factores de transcripción que podrían estar involucrados en la quimiodiversidad en Withania. El silenciamiento génico inducido por virus de la esterol ∆7-reductasa (WsDWF5) involucrada en la síntesis de colesterol 24-metileno, estructura principal de withanolida, sugiere el papel de esta enzima en la biosíntesis de withanolidas. La información generada, en este estudio, proporciona un recurso rico para el análisis funcional de genes específicos de withanolida para dilucidar la biosíntesis de withanolida específica del quimiotipo y del tejido. Este recurso genómico también ayudará en el desarrollo de nuevas herramientas para la genómica funcional y la cría en Withania.',\n",
              "      'descriptionType': 'Other'}],\n",
              "    'geoLocations': [],\n",
              "    'fundingReferences': [],\n",
              "    'url': 'https://gresis.osc.int//doi/10.60692/4jv94-r2r41',\n",
              "    'contentUrl': None,\n",
              "    'metadataVersion': 0,\n",
              "    'schemaVersion': 'http://datacite.org/schema/kernel-4',\n",
              "    'source': 'api',\n",
              "    'isActive': True,\n",
              "    'state': 'findable',\n",
              "    'reason': None,\n",
              "    'viewCount': 0,\n",
              "    'downloadCount': 0,\n",
              "    'referenceCount': 0,\n",
              "    'citationCount': 0,\n",
              "    'partCount': 0,\n",
              "    'partOfCount': 0,\n",
              "    'versionCount': 0,\n",
              "    'versionOfCount': 0,\n",
              "    'created': '2024-06-03T10:55:10Z',\n",
              "    'registered': '2024-06-03T10:55:10Z',\n",
              "    'published': None,\n",
              "    'updated': '2024-06-03T10:55:10Z'},\n",
              "   'relationships': {'client': {'data': {'id': 'iapx.bsycxq',\n",
              "      'type': 'clients'}}}},\n",
              "  {'id': '10.60692/0d102-xph18',\n",
              "   'type': 'dois',\n",
              "   'attributes': {'doi': '10.60692/0d102-xph18',\n",
              "    'identifiers': [{'identifier': 'https://openalex.org/W2058706754',\n",
              "      'identifierType': 'Other'}],\n",
              "    'creators': [{'name': 'Sumya Pathak',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Sumya Pathak',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Deepika Lakhwani',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Deepika Lakhwani',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Parul Gupta',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Parul Gupta',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-0190-8753',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Binod Mishra',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Binod Mishra',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0003-2364-6405',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Sudhir Shukla',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Sudhir Shukla',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0003-4510-0585',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Mehar Hasan Asif',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Mehar Hasan Asif',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-7640-4779',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Prabodh Kumar Trivedi',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Prabodh Kumar Trivedi',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0001-6463-1731',\n",
              "        'nameIdentifierScheme': 'ORCID'}]}],\n",
              "    'titles': [{'title': 'Comparative Transcriptome Analysis Using High Papaverine Mutant of Papaver somniferum Reveals Pathway and Uncharacterized Steps of Papaverine Biosynthesis'},\n",
              "     {'lang': 'ara',\n",
              "      'title': 'تحليل النسخ المقارن باستخدام مسخ البابافيرين العالي من البابافيرين يكشف عن المسار والخطوات غير المميزة للتركيب الحيوي للبابافيرين',\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'fra',\n",
              "      'title': \"L'analyse comparative du transcriptome à l'aide d'un mutant à haute teneur en papavérine de Papaver somniferum révèle la voie et les étapes non caractérisées de la biosynthèse de la papavérine\",\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'spa',\n",
              "      'title': 'El análisis comparativo del transcriptoma con mutante de papaverina alta de Papaver somniferum revela la vía y los pasos no caracterizados de la biosíntesis de papaverina',\n",
              "      'titleType': 'TranslatedTitle'}],\n",
              "    'publisher': 'OpenAlex',\n",
              "    'container': {},\n",
              "    'publicationYear': 2013,\n",
              "    'subjects': [{'subject': 'Therapeutic Potential of Berberine Alkaloids'},\n",
              "     {'subject': 'Pharmacology'},\n",
              "     {'subject': 'Medicine'},\n",
              "     {'subject': 'Health Sciences'},\n",
              "     {'subject': 'Pharmacology of Aconitum Alkaloids in Traditional Medicine'},\n",
              "     {'subject': 'Pharmacology, Toxicology and Pharmaceutics'},\n",
              "     {'subject': 'Life Sciences'},\n",
              "     {'subject': 'Pharmacology of Kratom Alkaloids and Related Compounds'},\n",
              "     {'subject': 'Pharmacokinetics'},\n",
              "     {'subject': 'Papaverine'},\n",
              "     {'subject': 'Benzylisoquinoline'},\n",
              "     {'subject': 'Papaver'},\n",
              "     {'subject': 'Opium Poppy'},\n",
              "     {'subject': 'Biosynthesis'},\n",
              "     {'subject': 'Transcriptome'},\n",
              "     {'subject': 'Mutant'},\n",
              "     {'subject': 'Sanguinarine'},\n",
              "     {'subject': 'Biology'},\n",
              "     {'subject': 'Biochemistry'},\n",
              "     {'subject': 'Gene'},\n",
              "     {'subject': 'Gene expression'},\n",
              "     {'subject': 'Alkaloid'},\n",
              "     {'subject': 'Botany'},\n",
              "     {'subject': 'Endocrinology'}],\n",
              "    'contributors': [],\n",
              "    'dates': [{'date': '2013-05-30', 'dateType': 'Issued'}],\n",
              "    'language': 'en',\n",
              "    'types': {'ris': 'RPRT',\n",
              "     'bibtex': 'article',\n",
              "     'citeproc': 'article-journal',\n",
              "     'schemaOrg': 'ScholarlyArticle',\n",
              "     'resourceType': '',\n",
              "     'resourceTypeGeneral': 'Text'},\n",
              "    'relatedIdentifiers': [{'relationType': 'HasVersion',\n",
              "      'relatedIdentifier': '10.60692/x6p91-gfv58',\n",
              "      'relatedIdentifierType': 'DOI'}],\n",
              "    'relatedItems': [],\n",
              "    'sizes': [],\n",
              "    'formats': [],\n",
              "    'version': 'v1',\n",
              "    'rightsList': [{'rights': 'cc-by'}],\n",
              "    'descriptions': [{'description': \"The benzylisoquinoline alkaloid papaverine, synthesized in low amount in most of the opium poppy varieties of Papaver somniferum, is used as a vasodilator muscle relaxant and antispasmodic. Papaverine biosynthesis remains controversial as two different routes utilizing either (S)-coclaurine or (S)-reticuline have been proposed with uncharacterized intermediate steps. In an attempt to elucidate papaverine biosynthesis and identify putative genes involved in uncharacterized steps, we carried out comparative transcriptome analysis of high papaverine mutant (pap1) and normal cultivar (BR086) of P. somniferum. This natural mutant synthesizes more than 12-fold papaverine in comparison to BR086. We established more than 238 Mb transcriptome data separately for pap1 and BR086. Assembly of reads generated 127,342 and 106,128 unigenes in pap1 and BR086, respectively. Digital gene expression analysis of transcriptomes revealed 3,336 differentially expressing unigenes. Enhanced expression of (S)-norcoclaurine-6-O-methyltransferase (6OMT), (S)-3'-hydroxy-N-methylcoclaurine 4'-O-methyltransferase (4'OMT), norreticuline 7-O-methyltransferase (N7OMT) and down-regulation of reticuline 7-O-methyltransferase (7OMT) in pap1 in comparison to BR086 suggest (S)-coclaurine as the route for papaverine biosynthesis. We also identified several methyltransferases and dehydrogenases with enhanced expression in pap1 in comparison to BR086. Our analysis using natural mutant, pap1, concludes that (S)-coclaurine is the branch-point intermediate and preferred route for papaverine biosynthesis. Differentially expressing methyltransferases and dehydrogenases identified in this study will help in elucidating complete biosynthetic pathway of papaverine. The information generated will be helpful in developing strategies for enhanced biosynthesis of papaverine through biotechnological approaches.\",\n",
              "      'descriptionType': 'Abstract'},\n",
              "     {'lang': 'ara',\n",
              "      'description': \"يستخدم البنزيليزوكينولين القلوي بابافيرين، الذي يتم تصنيعه بكمية منخفضة في معظم أصناف خشخاش الأفيون من الخشخاش المنوم، كمرخي للعضلات ومضاد للتشنج. لا يزال التخليق الحيوي للبابافيرين مثيرًا للجدل حيث تم اقتراح طريقين مختلفين باستخدام إما (S) - coclaurine أو (S) - reticuline بخطوات وسيطة غير محددة. في محاولة لتوضيح التخليق الحيوي للبابافيرين وتحديد الجينات المفترضة المشاركة في خطوات غير محددة، أجرينا تحليلًا مقارنًا لنسخة من متحور البابافيرين العالي (pap1) والصنف الطبيعي (BR086) من P. somniferum. تقوم هذه الطفرة الطبيعية بتصنيع بابافيرين أكثر من 12 ضعفًا مقارنة بـ BR086. أنشأنا أكثر من 238 ميغابايت من بيانات النسخ بشكل منفصل لـ pap1 و BR086. ولدت مجموعة القراءات 127،342 و 106،128 أحادي الجين في pap1 و BR086، على التوالي. كشف تحليل التعبير الجيني الرقمي للنسخ عن 3336 جينًا مختلفًا. التعبير المعزز عن ناقلة الميثيل (S) - norcoclaurine -6 - O - methyltransferase (6OMT)، (S)-3'- hydroxy - N -methylcoclaurine 4'- O - methyltransferase (4'OMT)، نوريتيكولين 7 - O - methyltransferase (N7OMT) والتنظيم السفلي للريكتولين 7 - O - methyltransferase (7OMT) في pap1 مقارنة بـ BR086 - coclaurine كمسار للتخليق الحيوي للبابافيرين. حددنا أيضًا العديد من ناقلات الميثيل ونازعات الهيدروجين مع تعبير محسن في pap1 مقارنة بـ BR086. يخلص تحليلنا باستخدام الطفرة الطبيعية، pap1، إلى أن (S)- coclaurine هو الطريق الوسيط والمفضل لنقطة الفرع للتخليق الحيوي للبابافيرين. سيساعد التعبير التفاضلي عن ناقلات الميثيل ونازعات الهيدروجين المحددة في هذه الدراسة في توضيح مسار التخليق الحيوي الكامل للبابافيرين. ستكون المعلومات التي يتم إنشاؤها مفيدة في تطوير استراتيجيات لتعزيز التخليق الحيوي للبابافيرين من خلال مناهج التكنولوجيا الحيوية.\",\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'fra',\n",
              "      'description': \"La papavérine alcaloïde benzylisoquinoléine, synthétisée en faible quantité dans la plupart des variétés de pavot à opium de Papaver somniferum, est utilisée comme relaxant musculaire vasodilatateur et antispasmodique. La biosynthèse de la papavérine reste controversée car deux voies différentes utilisant soit la (S)-coclaurine soit la (S)-réticuline ont été proposées avec des étapes intermédiaires non caractérisées. Dans le but d'élucider la biosynthèse de la papavérine et d'identifier les gènes putatifs impliqués dans des étapes non caractérisées, nous avons effectué une analyse comparative du transcriptome du mutant papavérine élevé (pap1) et du cultivar normal (BR086) de P. somniferum. Ce mutant naturel synthétise plus de 12 fois la papavérine par rapport à BR086. Nous avons établi plus de 238 Mb de données de transcriptome séparément pour pap1 et BR086. L'assemblage des lectures a généré 127 342 et 106 128 unigènes dans pap1 et BR086, respectivement. L'analyse de l'expression génique numérique des transcriptomes a révélé 3 336 unigènes exprimant de manière différentielle. L'expression améliorée de la (S) -norcoclaurine-6-O-méthyltransférase (6OMT), de la (S)-3'-hydroxy-N-méthylcoclaurine 4'-O-méthyltransférase (4'OMT), de la norréticuline 7-O-méthyltransférase (N7OMT) et la régulation négative de la réticuline 7-O-méthyltransférase (7OMT) dans pap1 par rapport à BR086 suggèrent la (S)-coclaurine comme voie de biosynthèse de la papavérine. Nous avons également identifié plusieurs méthyltransférases et déshydrogénases avec une expression accrue dans pap1 par rapport à BR086. Notre analyse utilisant le mutant naturel, pap1, conclut que la (S) -coclaurine est l'intermédiaire du point de branchement et la voie préférée pour la biosynthèse de la papavérine. L'expression différentielle des méthyltransférases et des déshydrogénases identifiées dans cette étude aidera à élucider la voie biosynthétique complète de la papavérine. Les informations générées seront utiles pour élaborer des stratégies visant à améliorer la biosynthèse de la papavérine grâce à des approches biotechnologiques.\",\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'spa',\n",
              "      'description': \"El alcaloide bencilisoquinolina papaverina, sintetizado en baja cantidad en la mayoría de las variedades de adormidera de Papaver somniferum, se utiliza como relajante muscular vasodilatador y antiespasmódico. La biosíntesis de papaverina sigue siendo controvertida, ya que se han propuesto dos rutas diferentes que utilizan (S) -coclaurinao (S) -reticulina con pasos intermedios no caracterizados. En un intento por dilucidar la biosíntesis de papaverina e identificar genes putativos involucrados en pasos no caracterizados, llevamos a cabo un análisis comparativo del transcriptoma del mutante con alto contenido de papaverina (pap1) y el cultivar normal (BR086) de P. somniferum. Este mutante natural sintetiza más de 12 veces papaverina en comparación con BR086. Establecimos más de 238 Mb de datos del transcriptoma por separado para pap1 y BR086. El ensamblaje de lecturas generó 127.342 y 106.128 unigenes en pap1 y BR086, respectivamente. El análisis de la expresión génica digital de los transcriptomas reveló 3.336 unigenes que expresan diferencialmente. La expresión mejorada de (S)-norcoclaurina-6-O-metiltransferasa (6OMT), (S)-3'-hidroxi-N-metilcoclaurina 4'-O-metiltransferasa (4'OMT), norreticulina 7-O-metiltransferasa (N7OMT) y la regulación negativa de reticulina 7-O-metiltransferasa (7OMT) en pap1 en comparación con BR086 sugieren (S) -coclaurina como la ruta para la biosíntesis de papaverina. También identificamos varias metiltransferasas y deshidrogenasas con expresión mejorada en pap1 en comparación con BR086. Nuestro análisis utilizando el mutante natural, pap1, concluye que la (S) -coclaurina es el punto intermedio de ramificación y la ruta preferida para la biosíntesis de papaverina. La expresión diferencial de metiltransferasas y deshidrogenasas identificadas en este estudio ayudará a dilucidar la ruta biosintética completa de la papaverina. La información generada será útil para desarrollar estrategias para mejorar la biosíntesis de papaverina a través de enfoques biotecnológicos.\",\n",
              "      'descriptionType': 'Other'}],\n",
              "    'geoLocations': [],\n",
              "    'fundingReferences': [],\n",
              "    'url': 'https://gresis.osc.int//doi/10.60692/0d102-xph18',\n",
              "    'contentUrl': None,\n",
              "    'metadataVersion': 0,\n",
              "    'schemaVersion': 'http://datacite.org/schema/kernel-4',\n",
              "    'source': 'api',\n",
              "    'isActive': True,\n",
              "    'state': 'findable',\n",
              "    'reason': None,\n",
              "    'viewCount': 0,\n",
              "    'downloadCount': 0,\n",
              "    'referenceCount': 0,\n",
              "    'citationCount': 0,\n",
              "    'partCount': 0,\n",
              "    'partOfCount': 0,\n",
              "    'versionCount': 0,\n",
              "    'versionOfCount': 0,\n",
              "    'created': '2024-06-03T10:47:28Z',\n",
              "    'registered': '2024-06-03T10:47:28Z',\n",
              "    'published': None,\n",
              "    'updated': '2024-06-03T10:47:28Z'},\n",
              "   'relationships': {'client': {'data': {'id': 'iapx.bsycxq',\n",
              "      'type': 'clients'}}}},\n",
              "  {'id': '10.60692/x6p91-gfv58',\n",
              "   'type': 'dois',\n",
              "   'attributes': {'doi': '10.60692/x6p91-gfv58',\n",
              "    'identifiers': [{'identifier': 'oai:gresis.org:x6p91-gfv58',\n",
              "      'identifierType': 'oai'},\n",
              "     {'identifier': 'https://openalex.org/W2058706754',\n",
              "      'identifierType': 'Other'}],\n",
              "    'creators': [{'name': 'Sumya Pathak',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Sumya Pathak',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Deepika Lakhwani',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Deepika Lakhwani',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Parul Gupta',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Parul Gupta',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-0190-8753',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Binod Mishra',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Binod Mishra',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0003-2364-6405',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Sudhir Shukla',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Sudhir Shukla',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0003-4510-0585',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Mehar Hasan Asif',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Mehar Hasan Asif',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-7640-4779',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Prabodh Kumar Trivedi',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Prabodh Kumar Trivedi',\n",
              "      'affiliation': ['National Botanical Research Institute'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0001-6463-1731',\n",
              "        'nameIdentifierScheme': 'ORCID'}]}],\n",
              "    'titles': [{'title': 'Comparative Transcriptome Analysis Using High Papaverine Mutant of Papaver somniferum Reveals Pathway and Uncharacterized Steps of Papaverine Biosynthesis'},\n",
              "     {'lang': 'ara',\n",
              "      'title': 'تحليل النسخ المقارن باستخدام مسخ البابافيرين العالي من البابافيرين يكشف عن المسار والخطوات غير المميزة للتركيب الحيوي للبابافيرين',\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'fra',\n",
              "      'title': \"L'analyse comparative du transcriptome à l'aide d'un mutant à haute teneur en papavérine de Papaver somniferum révèle la voie et les étapes non caractérisées de la biosynthèse de la papavérine\",\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'spa',\n",
              "      'title': 'El análisis comparativo del transcriptoma con mutante de papaverina alta de Papaver somniferum revela la vía y los pasos no caracterizados de la biosíntesis de papaverina',\n",
              "      'titleType': 'TranslatedTitle'}],\n",
              "    'publisher': 'OpenAlex',\n",
              "    'container': {},\n",
              "    'publicationYear': 2013,\n",
              "    'subjects': [{'subject': 'Therapeutic Potential of Berberine Alkaloids'},\n",
              "     {'subject': 'Pharmacology'},\n",
              "     {'subject': 'Medicine'},\n",
              "     {'subject': 'Health Sciences'},\n",
              "     {'subject': 'Pharmacology of Aconitum Alkaloids in Traditional Medicine'},\n",
              "     {'subject': 'Pharmacology, Toxicology and Pharmaceutics'},\n",
              "     {'subject': 'Life Sciences'},\n",
              "     {'subject': 'Pharmacology of Kratom Alkaloids and Related Compounds'},\n",
              "     {'subject': 'Pharmacokinetics'},\n",
              "     {'subject': 'Papaverine'},\n",
              "     {'subject': 'Benzylisoquinoline'},\n",
              "     {'subject': 'Papaver'},\n",
              "     {'subject': 'Opium Poppy'},\n",
              "     {'subject': 'Biosynthesis'},\n",
              "     {'subject': 'Transcriptome'},\n",
              "     {'subject': 'Mutant'},\n",
              "     {'subject': 'Sanguinarine'},\n",
              "     {'subject': 'Biology'},\n",
              "     {'subject': 'Biochemistry'},\n",
              "     {'subject': 'Gene'},\n",
              "     {'subject': 'Gene expression'},\n",
              "     {'subject': 'Alkaloid'},\n",
              "     {'subject': 'Botany'},\n",
              "     {'subject': 'Endocrinology'}],\n",
              "    'contributors': [],\n",
              "    'dates': [{'date': '2013-05-30', 'dateType': 'Issued'}],\n",
              "    'language': 'en',\n",
              "    'types': {'ris': 'RPRT',\n",
              "     'bibtex': 'article',\n",
              "     'citeproc': 'article-journal',\n",
              "     'schemaOrg': 'ScholarlyArticle',\n",
              "     'resourceType': '',\n",
              "     'resourceTypeGeneral': 'Text'},\n",
              "    'relatedIdentifiers': [{'relationType': 'IsVersionOf',\n",
              "      'relatedIdentifier': '10.60692/0d102-xph18',\n",
              "      'relatedIdentifierType': 'DOI'}],\n",
              "    'relatedItems': [],\n",
              "    'sizes': [],\n",
              "    'formats': [],\n",
              "    'version': 'v1',\n",
              "    'rightsList': [{'rights': 'cc-by'}],\n",
              "    'descriptions': [{'description': \"The benzylisoquinoline alkaloid papaverine, synthesized in low amount in most of the opium poppy varieties of Papaver somniferum, is used as a vasodilator muscle relaxant and antispasmodic. Papaverine biosynthesis remains controversial as two different routes utilizing either (S)-coclaurine or (S)-reticuline have been proposed with uncharacterized intermediate steps. In an attempt to elucidate papaverine biosynthesis and identify putative genes involved in uncharacterized steps, we carried out comparative transcriptome analysis of high papaverine mutant (pap1) and normal cultivar (BR086) of P. somniferum. This natural mutant synthesizes more than 12-fold papaverine in comparison to BR086. We established more than 238 Mb transcriptome data separately for pap1 and BR086. Assembly of reads generated 127,342 and 106,128 unigenes in pap1 and BR086, respectively. Digital gene expression analysis of transcriptomes revealed 3,336 differentially expressing unigenes. Enhanced expression of (S)-norcoclaurine-6-O-methyltransferase (6OMT), (S)-3'-hydroxy-N-methylcoclaurine 4'-O-methyltransferase (4'OMT), norreticuline 7-O-methyltransferase (N7OMT) and down-regulation of reticuline 7-O-methyltransferase (7OMT) in pap1 in comparison to BR086 suggest (S)-coclaurine as the route for papaverine biosynthesis. We also identified several methyltransferases and dehydrogenases with enhanced expression in pap1 in comparison to BR086. Our analysis using natural mutant, pap1, concludes that (S)-coclaurine is the branch-point intermediate and preferred route for papaverine biosynthesis. Differentially expressing methyltransferases and dehydrogenases identified in this study will help in elucidating complete biosynthetic pathway of papaverine. The information generated will be helpful in developing strategies for enhanced biosynthesis of papaverine through biotechnological approaches.\",\n",
              "      'descriptionType': 'Abstract'},\n",
              "     {'lang': 'ara',\n",
              "      'description': \"يستخدم البنزيليزوكينولين القلوي بابافيرين، الذي يتم تصنيعه بكمية منخفضة في معظم أصناف خشخاش الأفيون من الخشخاش المنوم، كمرخي للعضلات ومضاد للتشنج. لا يزال التخليق الحيوي للبابافيرين مثيرًا للجدل حيث تم اقتراح طريقين مختلفين باستخدام إما (S) - coclaurine أو (S) - reticuline بخطوات وسيطة غير محددة. في محاولة لتوضيح التخليق الحيوي للبابافيرين وتحديد الجينات المفترضة المشاركة في خطوات غير محددة، أجرينا تحليلًا مقارنًا لنسخة من متحور البابافيرين العالي (pap1) والصنف الطبيعي (BR086) من P. somniferum. تقوم هذه الطفرة الطبيعية بتصنيع بابافيرين أكثر من 12 ضعفًا مقارنة بـ BR086. أنشأنا أكثر من 238 ميغابايت من بيانات النسخ بشكل منفصل لـ pap1 و BR086. ولدت مجموعة القراءات 127،342 و 106،128 أحادي الجين في pap1 و BR086، على التوالي. كشف تحليل التعبير الجيني الرقمي للنسخ عن 3336 جينًا مختلفًا. التعبير المعزز عن ناقلة الميثيل (S) - norcoclaurine -6 - O - methyltransferase (6OMT)، (S)-3'- hydroxy - N -methylcoclaurine 4'- O - methyltransferase (4'OMT)، نوريتيكولين 7 - O - methyltransferase (N7OMT) والتنظيم السفلي للريكتولين 7 - O - methyltransferase (7OMT) في pap1 مقارنة بـ BR086 - coclaurine كمسار للتخليق الحيوي للبابافيرين. حددنا أيضًا العديد من ناقلات الميثيل ونازعات الهيدروجين مع تعبير محسن في pap1 مقارنة بـ BR086. يخلص تحليلنا باستخدام الطفرة الطبيعية، pap1، إلى أن (S)- coclaurine هو الطريق الوسيط والمفضل لنقطة الفرع للتخليق الحيوي للبابافيرين. سيساعد التعبير التفاضلي عن ناقلات الميثيل ونازعات الهيدروجين المحددة في هذه الدراسة في توضيح مسار التخليق الحيوي الكامل للبابافيرين. ستكون المعلومات التي يتم إنشاؤها مفيدة في تطوير استراتيجيات لتعزيز التخليق الحيوي للبابافيرين من خلال مناهج التكنولوجيا الحيوية.\",\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'fra',\n",
              "      'description': \"La papavérine alcaloïde benzylisoquinoléine, synthétisée en faible quantité dans la plupart des variétés de pavot à opium de Papaver somniferum, est utilisée comme relaxant musculaire vasodilatateur et antispasmodique. La biosynthèse de la papavérine reste controversée car deux voies différentes utilisant soit la (S)-coclaurine soit la (S)-réticuline ont été proposées avec des étapes intermédiaires non caractérisées. Dans le but d'élucider la biosynthèse de la papavérine et d'identifier les gènes putatifs impliqués dans des étapes non caractérisées, nous avons effectué une analyse comparative du transcriptome du mutant papavérine élevé (pap1) et du cultivar normal (BR086) de P. somniferum. Ce mutant naturel synthétise plus de 12 fois la papavérine par rapport à BR086. Nous avons établi plus de 238 Mb de données de transcriptome séparément pour pap1 et BR086. L'assemblage des lectures a généré 127 342 et 106 128 unigènes dans pap1 et BR086, respectivement. L'analyse de l'expression génique numérique des transcriptomes a révélé 3 336 unigènes exprimant de manière différentielle. L'expression améliorée de la (S) -norcoclaurine-6-O-méthyltransférase (6OMT), de la (S)-3'-hydroxy-N-méthylcoclaurine 4'-O-méthyltransférase (4'OMT), de la norréticuline 7-O-méthyltransférase (N7OMT) et la régulation négative de la réticuline 7-O-méthyltransférase (7OMT) dans pap1 par rapport à BR086 suggèrent la (S)-coclaurine comme voie de biosynthèse de la papavérine. Nous avons également identifié plusieurs méthyltransférases et déshydrogénases avec une expression accrue dans pap1 par rapport à BR086. Notre analyse utilisant le mutant naturel, pap1, conclut que la (S) -coclaurine est l'intermédiaire du point de branchement et la voie préférée pour la biosynthèse de la papavérine. L'expression différentielle des méthyltransférases et des déshydrogénases identifiées dans cette étude aidera à élucider la voie biosynthétique complète de la papavérine. Les informations générées seront utiles pour élaborer des stratégies visant à améliorer la biosynthèse de la papavérine grâce à des approches biotechnologiques.\",\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'spa',\n",
              "      'description': \"El alcaloide bencilisoquinolina papaverina, sintetizado en baja cantidad en la mayoría de las variedades de adormidera de Papaver somniferum, se utiliza como relajante muscular vasodilatador y antiespasmódico. La biosíntesis de papaverina sigue siendo controvertida, ya que se han propuesto dos rutas diferentes que utilizan (S) -coclaurinao (S) -reticulina con pasos intermedios no caracterizados. En un intento por dilucidar la biosíntesis de papaverina e identificar genes putativos involucrados en pasos no caracterizados, llevamos a cabo un análisis comparativo del transcriptoma del mutante con alto contenido de papaverina (pap1) y el cultivar normal (BR086) de P. somniferum. Este mutante natural sintetiza más de 12 veces papaverina en comparación con BR086. Establecimos más de 238 Mb de datos del transcriptoma por separado para pap1 y BR086. El ensamblaje de lecturas generó 127.342 y 106.128 unigenes en pap1 y BR086, respectivamente. El análisis de la expresión génica digital de los transcriptomas reveló 3.336 unigenes que expresan diferencialmente. La expresión mejorada de (S)-norcoclaurina-6-O-metiltransferasa (6OMT), (S)-3'-hidroxi-N-metilcoclaurina 4'-O-metiltransferasa (4'OMT), norreticulina 7-O-metiltransferasa (N7OMT) y la regulación negativa de reticulina 7-O-metiltransferasa (7OMT) en pap1 en comparación con BR086 sugieren (S) -coclaurina como la ruta para la biosíntesis de papaverina. También identificamos varias metiltransferasas y deshidrogenasas con expresión mejorada en pap1 en comparación con BR086. Nuestro análisis utilizando el mutante natural, pap1, concluye que la (S) -coclaurina es el punto intermedio de ramificación y la ruta preferida para la biosíntesis de papaverina. La expresión diferencial de metiltransferasas y deshidrogenasas identificadas en este estudio ayudará a dilucidar la ruta biosintética completa de la papaverina. La información generada será útil para desarrollar estrategias para mejorar la biosíntesis de papaverina a través de enfoques biotecnológicos.\",\n",
              "      'descriptionType': 'Other'}],\n",
              "    'geoLocations': [],\n",
              "    'fundingReferences': [],\n",
              "    'url': 'https://gresis.osc.int//doi/10.60692/x6p91-gfv58',\n",
              "    'contentUrl': None,\n",
              "    'metadataVersion': 0,\n",
              "    'schemaVersion': 'http://datacite.org/schema/kernel-4',\n",
              "    'source': 'api',\n",
              "    'isActive': True,\n",
              "    'state': 'findable',\n",
              "    'reason': None,\n",
              "    'viewCount': 0,\n",
              "    'downloadCount': 0,\n",
              "    'referenceCount': 0,\n",
              "    'citationCount': 0,\n",
              "    'partCount': 0,\n",
              "    'partOfCount': 0,\n",
              "    'versionCount': 0,\n",
              "    'versionOfCount': 0,\n",
              "    'created': '2024-06-03T10:47:28Z',\n",
              "    'registered': '2024-06-03T10:47:28Z',\n",
              "    'published': None,\n",
              "    'updated': '2024-06-03T10:47:28Z'},\n",
              "   'relationships': {'client': {'data': {'id': 'iapx.bsycxq',\n",
              "      'type': 'clients'}}}},\n",
              "  {'id': '10.60692/m626c-0zr93',\n",
              "   'type': 'dois',\n",
              "   'attributes': {'doi': '10.60692/m626c-0zr93',\n",
              "    'identifiers': [{'identifier': 'https://openalex.org/W2154398258',\n",
              "      'identifierType': 'Other'}],\n",
              "    'creators': [{'name': 'Anchal Garg',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Anchal Garg',\n",
              "      'affiliation': ['Central Institute of Medicinal and Aromatic Plants',\n",
              "       'Council of Scientific and Industrial Research'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Lalit Agrawal',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Lalit Agrawal',\n",
              "      'affiliation': ['National Botanical Research Institute',\n",
              "       'Council of Scientific and Industrial Research'],\n",
              "      'nameIdentifiers': []},\n",
              "     {'name': 'Rajesh Chandra Misra',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Rajesh Chandra Misra',\n",
              "      'affiliation': ['Central Institute of Medicinal and Aromatic Plants',\n",
              "       'Council of Scientific and Industrial Research'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0001-9594-3789',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Shubha Rani Sharma',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Shubha Rani Sharma',\n",
              "      'affiliation': ['Central Institute of Medicinal and Aromatic Plants',\n",
              "       'Council of Scientific and Industrial Research'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-9201-3532',\n",
              "        'nameIdentifierScheme': 'ORCID'}]},\n",
              "     {'name': 'Sumit Ghosh',\n",
              "      'nameType': 'Personal',\n",
              "      'familyName': 'Sumit Ghosh',\n",
              "      'affiliation': ['Central Institute of Medicinal and Aromatic Plants',\n",
              "       'Council of Scientific and Industrial Research'],\n",
              "      'nameIdentifiers': [{'nameIdentifier': '0000-0002-0767-6121',\n",
              "        'nameIdentifierScheme': 'ORCID'}]}],\n",
              "    'titles': [{'title': 'Andrographis paniculata transcriptome provides molecular insights into tissue-specific accumulation of medicinal diterpenes'},\n",
              "     {'lang': 'ara',\n",
              "      'title': 'يوفر أندروغرافيس بانيكولاتا ترانسكريبتوم رؤى جزيئية حول التراكم النوعي للأنسجة للديتربينات الطبية',\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'fra',\n",
              "      'title': \"Le transcriptome d'Andrographis paniculata fournit des informations moléculaires sur l'accumulation tissulaire spécifique de diterpènes médicinaux\",\n",
              "      'titleType': 'TranslatedTitle'},\n",
              "     {'lang': 'spa',\n",
              "      'title': 'El transcriptoma de Andrographis paniculata proporciona información molecular sobre la acumulación específica de tejido de diterpenos medicinales',\n",
              "      'titleType': 'TranslatedTitle'}],\n",
              "    'publisher': 'OpenAlex',\n",
              "    'container': {},\n",
              "    'publicationYear': 2015,\n",
              "    'subjects': [{'subject': 'Medicinal Properties of Andrographis Paniculata'},\n",
              "     {'subject': 'Complementary and alternative medicine'},\n",
              "     {'subject': 'Medicine'},\n",
              "     {'subject': 'Health Sciences'},\n",
              "     {'subject': 'Biosynthesis and Engineering of Terpenoids'},\n",
              "     {'subject': 'Molecular Biology'},\n",
              "     {'subject': 'Biochemistry, Genetics and Molecular Biology'},\n",
              "     {'subject': 'Life Sciences'},\n",
              "     {'subject': 'Molecular Mechanisms and Medical Applications of Ginseng'},\n",
              "     {'subject': 'Andrographolide'},\n",
              "     {'subject': 'Andrographis Paniculata'},\n",
              "     {'subject': 'Biology'},\n",
              "     {'subject': 'KEGG'},\n",
              "     {'subject': 'Transcriptome'},\n",
              "     {'subject': 'Andrographis paniculata'},\n",
              "     {'subject': 'Biochemistry'},\n",
              "     {'subject': 'Gene'},\n",
              "     {'subject': 'Genetics'},\n",
              "     {'subject': 'FOS: Biological sciences',\n",
              "      'schemeUri': 'http://www.oecd.org/science/inno/38235147.pdf',\n",
              "      'subjectScheme': 'Fields of Science and Technology (FOS)'},\n",
              "     {'subject': 'Gene expression'},\n",
              "     {'subject': 'Alternative medicine'},\n",
              "     {'subject': 'Pathology'}],\n",
              "    'contributors': [],\n",
              "    'dates': [{'date': '2015-09-02', 'dateType': 'Issued'}],\n",
              "    'language': 'en',\n",
              "    'types': {'ris': 'RPRT',\n",
              "     'bibtex': 'article',\n",
              "     'citeproc': 'article-journal',\n",
              "     'schemaOrg': 'ScholarlyArticle',\n",
              "     'resourceType': '',\n",
              "     'resourceTypeGeneral': 'Text'},\n",
              "    'relatedIdentifiers': [{'relationType': 'HasVersion',\n",
              "      'relatedIdentifier': '10.60692/et4wr-1aj51',\n",
              "      'relatedIdentifierType': 'DOI'}],\n",
              "    'relatedItems': [],\n",
              "    'sizes': [],\n",
              "    'formats': [],\n",
              "    'version': 'v1',\n",
              "    'rightsList': [{'rights': 'cc-by'}],\n",
              "    'descriptions': [{'description': 'Kalmegh (Andrographis paniculata) has been widely exploited in traditional medicine for the treatment of infectious diseases and health disorders. Ent-labdane-related diterpene (ent-LRD) specialized (i.e., secondary) metabolites of kalmegh such as andrographolide, neoandrographolide and 14-deoxy-11,12-didehydroandrographolide, are known for variety of pharmacological activities. However, due to the lack of genomic and transcriptomic information, underlying molecular basis of ent-LRDs biosynthesis has remained largely unknown. To identify candidate genes of the ent-LRD biosynthetic pathway, we performed comparative transcriptome analysis using leaf and root tissues that differentially accumulate ent-LRDs.De novo assembly of Illumina HiSeq2000 platform-generated paired-end sequencing reads resulted into 69,011 leaf and 64,244 root transcripts which were assembled into a total of 84,628 unique transcripts. Annotation of these transcripts to the Uniprot, Kyoto Encyclopedia of Genes and Genomes (KEGG) and Carbohydrate-Active Enzymes (CAZy) databases identified candidate transcripts of the ent-LRD biosynthetic pathway. These included transcripts that encode enzymes of the plastidial 2C-methyl-D-erythritol-4-phosphate pathway which provides C5 isoprenoid precursors for the ent-LRDs biosynthesis, geranylgeranyl diphosphate synthase, class II diterpene synthase (diTPS), cytochrome P450 monooxygenase and glycosyltransferase. Three class II diTPSs (ApCPS1, ApCPS2 and ApCPS3) that showed distinct tissue-specific expression profiles and are phylogenetically related to the dicotyledon ent-copalyl diphosphate synthases, are identified. ApCPS1, ApCPS2 and ApCPS3 encode for 832-, 817- and 797- amino acids proteins of 55-63 % identity, respectively. Spatio-temporal patterns of transcripts and ent-LRDs accumulation are consistent with the involvement of ApCPS1 in general (i.e., primary) metabolism for the biosynthesis of phytohormone gibberellin, ApCPS2 in leaf specialized ent-LRDs biosynthesis and ApCPS3 in root diterpene biosynthesis. Moreover, simple sequence repeats (SSRs) that might assist in genotyping and developing specific chemotypes were identified in transcripts of the specialized metabolic pathways, including ent-LRDs.Comparative analysis of root and leaf transcriptomes disclosed novel genes of the ent-LRD biosynthetic pathway, including three class II diTPSs that showed discrete spatio-temporal expression patterns; thus, suggesting their participation into distinct diterpene metabolic pathways of kalmegh. Overall, these results will be useful in understanding molecular basis of the medicinal ent-LRDs biosynthesis and developing breeding strategies for improving their yields.',\n",
              "      'descriptionType': 'Abstract'},\n",
              "     {'lang': 'ara',\n",
              "      'description': 'تم استغلال Kalmegh (Andrographis paniculata) على نطاق واسع في الطب التقليدي لعلاج الأمراض المعدية والاضطرابات الصحية. تشتهر المستقلبات المتخصصة (أي الثانوية) للديتربين المرتبط باللبدان (ent - LRD) للكالميغ مثل الأندروغرافوليد والنيوأندروغرافوليد و 14 -ديوكسي-11 و 12 -ثنائي هيدرو الأندروغرافوليد بمجموعة متنوعة من الأنشطة الدوائية. ومع ذلك، نظرًا لنقص المعلومات الجينية والنسخية، ظل الأساس الجزيئي الأساسي للتخليق الحيوي لـ ent - LRDs غير معروف إلى حد كبير. لتحديد الجينات المرشحة لمسار التخليق الحيوي ent - LRD، أجرينا تحليل transcriptome المقارن باستخدام أنسجة الأوراق والجذور التي تتراكم بشكل تفاضلي ent - LRDs. أدى التجميع الجديد لقراءات تسلسل النهاية المزدوجة التي تم إنشاؤها بواسطة منصة Illumina HiSeq2000 إلى 69,011 ورقة و 64,244 نسخة جذر تم تجميعها في إجمالي 84,628 نسخة فريدة. حدد شرح هذه النصوص على قواعد بيانات Uniprot و Kyoto Encyclopedia of Genes and Genomes (KEGG) و Carbohydrate - Active Enzymes (CAZy) النصوص المرشحة لمسار التخليق الحيوي ent - LRD. وشملت هذه النسخ التي تشفر إنزيمات المسار البلاستيدي 2C - methyl - D - erythritol -4 - phosphate الذي يوفر سلائف إيزوبرينويد C5 للتخليق الحيوي ent - LRDs، سينثاز ثنائي فوسفات الجيرانيل، سينثاز ديتربين من الفئة الثانية (diTPS)، السيتوكروم P450 أحادي الأكسجين وناقلات الجليكوزيل. تم تحديد ثلاثة أنواع من مركبات ثنائي TPS من الفئة الثانية (ApCPS1 و ApCPS2 و ApCPS3) التي أظهرت ملامح تعبير مميزة خاصة بالأنسجة وترتبط من الناحية الوراثية بمركبات ثنائي فلويدون إنت كوباليل ثنائي الفوسفات. ترميز ApCPS1 و ApCPS2 و ApCPS3 لبروتينات الأحماض الأمينية 832 و 817 و 797 ذات هوية 55-63 ٪ على التوالي. تتوافق الأنماط المكانية والزمانية للنصوص وتراكم ent - LRDs مع مشاركة ApCPS1 في التمثيل الغذائي العام (أي الأولي) للتخليق الحيوي للهرمون النباتي gibberellin، ApCPS2 في التخليق الحيوي المتخصص لـ ent - LRDs و ApCPS3 في التخليق الحيوي للديتربين الجذري. علاوة على ذلك، تم تحديد التكرارات التسلسلية البسيطة (SSRs) التي قد تساعد في التنميط الجيني وتطوير أنماط كيميائية محددة في نصوص المسارات الأيضية المتخصصة، بما في ذلك ent - LRDs. كشف التحليل المقارن لنسخ الجذر والأوراق عن جينات جديدة لمسار التخليق الحيوي ent - LRD، بما في ذلك ثلاثة أنواع من diTPSs من الفئة الثانية أظهرت أنماط تعبير مكاني- زماني منفصلة ؛ وبالتالي، اقتراح مشاركتهم في مسارات أيضية مميزة للديتربين في كالميغ. بشكل عام، ستكون هذه النتائج مفيدة في فهم الأساس الجزيئي للتخليق الحيوي لـ ent - LRDs الطبي وتطوير استراتيجيات التكاثر لتحسين غلاتها.',\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'fra',\n",
              "      'description': \"Le kalmegh (Andrographis paniculata) a été largement exploité en médecine traditionnelle pour le traitement des maladies infectieuses et des troubles de la santé. Les métabolites spécialisés (c'est-à-dire secondaires) du diterpène lié à l'ent-labdane (ent-LRD) de kalmegh tels que l'andrographolide, le néoandrographolide et le 14-désoxy-11,12-didéhydroandrographolide, sont connus pour diverses activités pharmacologiques. Cependant, en raison du manque d'informations génomiques et transcriptomiques, la base moléculaire sous-jacente de la biosynthèse des ent-LRD est restée largement inconnue. Pour identifier les gènes candidats de la voie biosynthétique ent-LRD, nous avons effectué une analyse comparative du transcriptome à l'aide de tissus foliaires et racinaires qui accumulent différemment ent-LRD. L'assemblage novo des lectures de séquençage à extrémités appariées générées par la plate-forme Illumina HiSeq2000 a donné 69 011 transcrits foliaires et 64 244 transcrits racinaires qui ont été assemblés en un total de 84 628 transcrits uniques. L'annotation de ces transcriptions dans les bases de données Uniprot, Kyoto Encyclopedia of Genes and Genomes (KEGG) et Carbohydrate-Active Enzymes (CAZy) a identifié des transcriptions candidates de la voie biosynthétique ent-LRD. Ceux-ci comprenaient des transcrits qui codent pour des enzymes de la voie plastidiale 2C-méthyl-D-érythritol-4-phosphate qui fournit des précurseurs isoprénoïdes C5 pour la biosynthèse des ent-LRD, la géranylgéranyl diphosphate synthase, la diterpène synthase de classe II (diTPS), la monooxygénase du cytochrome P450 et la glycosyltransférase. Trois diTPS de classe II (ApCPS1, ApCPS2 et ApCPS3) qui présentaient des profils d'expression tissulaires spécifiques distincts et qui sont phylogénétiquement liés aux ent-copalyl diphosphate synthases dicotylédones, sont identifiés. ApCPS1, ApCPS2 et ApCPS3 codent pour des protéines d'acides aminés 832-, 817- et 797- de 55-63\\xa0% d'identité, respectivement. Les profils spatio-temporels des transcrits et de l'accumulation des ent-LRD sont compatibles avec l'implication de l'ApCPS1 dans le métabolisme général (c.-à-d. primaire) pour la biosynthèse de la phytohormone gibbérelline, de l'ApCPS2 dans la biosynthèse des ent-LRD spécialisés dans les feuilles et de l'ApCPS3 dans la biosynthèse des diterpènes racinaires. De plus, des répétitions de séquences simples (SSR) qui pourraient aider au génotypage et au développement de chimiotypes spécifiques ont été identifiées dans les transcrits des voies métaboliques spécialisées, y compris ent-LRD. L'analyse comparative des transcriptomes racinaires et foliaires a révélé de nouveaux gènes de la voie biosynthétique ent-LRD, y compris trois diTPS de classe II qui présentaient des modèles d'expression spatio-temporelle discrets\\xa0; suggérant ainsi leur participation à des voies métaboliques diterpéniques distinctes de kalmegh. Dans l'ensemble, ces résultats seront utiles pour comprendre la base moléculaire de la biosynthèse médicinale des ent-LRD et développer des stratégies de sélection pour améliorer leurs rendements.\",\n",
              "      'descriptionType': 'Other'},\n",
              "     {'lang': 'spa',\n",
              "      'description': 'Kalmegh (Andrographis paniculata) ha sido ampliamente explotado en la medicina tradicional para el tratamiento de enfermedades infecciosas y trastornos de salud. Los metabolitos especializados (es decir, secundarios) de diterpeno relacionados con ent-labdano (ent-LRD) de kalmegh tales como andrografólido, neoandrografólido y 14-desoxi-11,12-didehidroandrografólido, son conocidos por una variedad de actividades farmacológicas. Sin embargo, debido a la falta de información genómica y transcriptómica, la base molecular subyacente de la biosíntesis de ent-LRD sigue siendo en gran medida desconocida. Para identificar los genes candidatos de la vía biosintética ent-LRD, realizamos un análisis comparativo del transcriptoma utilizando tejidos de hojas y raíces que acumulan diferencialmente ent-LRD. El ensamblaje de novo de las lecturas de secuenciación de extremos emparejados generadas por la plataforma Illumina HiSeq2000 dio como resultado 69.011 transcritos de hojas y 64.244 de raíces que se ensamblaron en un total de 84.628 transcritos únicos. La anotación de estos transcritos en las bases de datos Uniprot, Kyoto Encyclopedia of Genes and Genomes (KEGG) y Carbohydrate-Active Enzymes (CAZy) identificó transcritos candidatos de la vía biosintética ent-LRD. Estos incluyeron transcritos que codifican enzimas de la vía plastidial 2C-metil-D-eritritol-4-fosfato que proporciona precursores de isoprenoides C5 para la biosíntesis de ent-LRD, geranilgeranil difosfato sintasa, diterpeno sintasa de clase II (diTPS), citocromo P450 monooxigenasa y glicosiltransferasa. Se identifican tres diTPS de clase II (ApCPS1, ApCPS2 y ApCPS3) que mostraron distintos perfiles de expresión específicos de tejido y están filogenéticamente relacionados con las dicotiledóneas ent-copalil difosfato sintasas. ApCPS1, ApCPS2 y ApCPS3 codifican proteínas de 832, 817 y 797 aminoácidos de 55-63 % de identidad, respectivamente. Los patrones espacio-temporales de los transcritos y la acumulación de ent-LRD son consistentes con la participación de ApCPS1 en el metabolismo general (es decir, primario) para la biosíntesis de la fitohormona giberelina, ApCPS2 en la biosíntesis de ent-LRD especializados en hojas y ApCPS3 en la biosíntesis de diterpenos de la raíz. Además, se identificaron repeticiones de secuencia simples (SSR) que podrían ayudar a genotipar y desarrollar quimiotipos específicos en transcritos de las vías metabólicas especializadas, incluyendo ent-LRD. El análisis comparativo de transcriptomos de raíz y hoja reveló genes novedosos de la vía biosintética ent-LRD, incluyendo tres diTPS de clase II que mostraron patrones de expresión espacio-temporales discretos; lo que sugiere su participación en distintas vías metabólicas de diterpenos de kalmegh. En general, estos resultados serán útiles para comprender las bases moleculares de la biosíntesis de ent-LRD medicinales y desarrollar estrategias de mejoramiento para mejorar sus rendimientos.',\n",
              "      'descriptionType': 'Other'}],\n",
              "    'geoLocations': [],\n",
              "    'fundingReferences': [],\n",
              "    'url': 'https://gresis.osc.int//doi/10.60692/m626c-0zr93',\n",
              "    'contentUrl': None,\n",
              "    'metadataVersion': 0,\n",
              "    'schemaVersion': 'http://datacite.org/schema/kernel-4',\n",
              "    'source': 'api',\n",
              "    'isActive': True,\n",
              "    'state': 'findable',\n",
              "    'reason': None,\n",
              "    'viewCount': 0,\n",
              "    'downloadCount': 0,\n",
              "    'referenceCount': 0,\n",
              "    'citationCount': 0,\n",
              "    'partCount': 0,\n",
              "    'partOfCount': 0,\n",
              "    'versionCount': 0,\n",
              "    'versionOfCount': 0,\n",
              "    'created': '2024-06-03T10:25:23Z',\n",
              "    'registered': '2024-06-03T10:25:23Z',\n",
              "    'published': None,\n",
              "    'updated': '2024-06-03T10:25:23Z'},\n",
              "   'relationships': {'client': {'data': {'id': 'iapx.bsycxq',\n",
              "      'type': 'clients'}}}}],\n",
              " 'meta': {'total': 8946,\n",
              "  'totalPages': 895,\n",
              "  'page': 1,\n",
              "  'states': [{'id': 'findable', 'title': 'Findable', 'count': 8946}],\n",
              "  'resourceTypes': [{'id': 'dataset', 'title': 'Dataset', 'count': 3434},\n",
              "   {'id': 'collection', 'title': 'Collection', 'count': 2293},\n",
              "   {'id': 'text', 'title': 'Text', 'count': 2017},\n",
              "   {'id': 'image', 'title': 'Image', 'count': 469},\n",
              "   {'id': 'journal-article', 'title': 'Journal Article', 'count': 186},\n",
              "   {'id': 'software', 'title': 'Software', 'count': 160},\n",
              "   {'id': 'other', 'title': 'Other', 'count': 124},\n",
              "   {'id': 'dissertation', 'title': 'Dissertation', 'count': 69},\n",
              "   {'id': 'audiovisual', 'title': 'Audiovisual', 'count': 62},\n",
              "   {'id': 'preprint', 'title': 'Preprint', 'count': 21},\n",
              "   {'id': 'workflow', 'title': 'Workflow', 'count': 14},\n",
              "   {'id': 'peer-review', 'title': 'Peer Review', 'count': 10},\n",
              "   {'id': 'output-management-plan',\n",
              "    'title': 'Output Management Plan',\n",
              "    'count': 9},\n",
              "   {'id': 'project', 'title': 'Project', 'count': 5},\n",
              "   {'id': 'conference-paper', 'title': 'Conference Paper', 'count': 3},\n",
              "   {'id': 'data-paper', 'title': 'Data Paper', 'count': 3},\n",
              "   {'id': 'report', 'title': 'Report', 'count': 2},\n",
              "   {'id': 'interactive-resource',\n",
              "    'title': 'Interactive Resource',\n",
              "    'count': 1}],\n",
              "  'created': [{'id': '2024', 'title': '2024', 'count': 632},\n",
              "   {'id': '2023', 'title': '2023', 'count': 850},\n",
              "   {'id': '2022', 'title': '2022', 'count': 1156},\n",
              "   {'id': '2021', 'title': '2021', 'count': 1171},\n",
              "   {'id': '2020', 'title': '2020', 'count': 1087},\n",
              "   {'id': '2019', 'title': '2019', 'count': 1218},\n",
              "   {'id': '2018', 'title': '2018', 'count': 844},\n",
              "   {'id': '2017', 'title': '2017', 'count': 1068},\n",
              "   {'id': '2016', 'title': '2016', 'count': 619},\n",
              "   {'id': '2015', 'title': '2015', 'count': 127}],\n",
              "  'published': [{'id': '2024', 'title': '2024', 'count': 456},\n",
              "   {'id': '2023', 'title': '2023', 'count': 867},\n",
              "   {'id': '2022', 'title': '2022', 'count': 1166},\n",
              "   {'id': '2021', 'title': '2021', 'count': 1091},\n",
              "   {'id': '2020', 'title': '2020', 'count': 1035},\n",
              "   {'id': '2019', 'title': '2019', 'count': 1224},\n",
              "   {'id': '2018', 'title': '2018', 'count': 795},\n",
              "   {'id': '2017', 'title': '2017', 'count': 762},\n",
              "   {'id': '2016', 'title': '2016', 'count': 659},\n",
              "   {'id': '2015', 'title': '2015', 'count': 449},\n",
              "   {'id': '2014', 'title': '2014', 'count': 107},\n",
              "   {'id': '2013', 'title': '2013', 'count': 80},\n",
              "   {'id': '2012', 'title': '2012', 'count': 69},\n",
              "   {'id': '2011', 'title': '2011', 'count': 142},\n",
              "   {'id': '2010', 'title': '2010', 'count': 14}],\n",
              "  'registered': [{'id': '2024', 'title': '2024', 'count': 635},\n",
              "   {'id': '2023', 'title': '2023', 'count': 849},\n",
              "   {'id': '2022', 'title': '2022', 'count': 1158},\n",
              "   {'id': '2021', 'title': '2021', 'count': 1169},\n",
              "   {'id': '2020', 'title': '2020', 'count': 1085},\n",
              "   {'id': '2019', 'title': '2019', 'count': 1218},\n",
              "   {'id': '2018', 'title': '2018', 'count': 843},\n",
              "   {'id': '2017', 'title': '2017', 'count': 1067},\n",
              "   {'id': '2016', 'title': '2016', 'count': 619},\n",
              "   {'id': '2015', 'title': '2015', 'count': 127}],\n",
              "  'providers': [{'id': 'otjm', 'title': 'Figshare Internal', 'count': 5641},\n",
              "   {'id': 'cern',\n",
              "    'title': 'CERN - European Organization for Nuclear Research',\n",
              "    'count': 739},\n",
              "   {'id': 'dryad', 'title': 'Dryad', 'count': 579},\n",
              "   {'id': 'iapx',\n",
              "    'title': 'Organisation of Southern Cooperation (OSC)',\n",
              "    'count': 226},\n",
              "   {'id': 'lpsw', 'title': 'University of Cambridge', 'count': 176},\n",
              "   {'id': 'cngb', 'title': 'China National GeneBank', 'count': 154},\n",
              "   {'id': 'elsevier', 'title': 'Elsevier (Mendeley Data)', 'count': 149},\n",
              "   {'id': 'cizs', 'title': 'UC San Diego', 'count': 92},\n",
              "   {'id': 'ubc', 'title': 'University of British Columbia', 'count': 76},\n",
              "   {'id': 'stdp', 'title': 'ETH Zurich', 'count': 73}],\n",
              "  'clients': [{'id': 'figshare.ars',\n",
              "    'title': 'figshare Academic Research System',\n",
              "    'count': 5641},\n",
              "   {'id': 'cern.zenodo', 'title': 'Zenodo', 'count': 739},\n",
              "   {'id': 'dryad.dryad', 'title': 'DRYAD', 'count': 579},\n",
              "   {'id': 'iapx.bsycxq',\n",
              "    'title': 'Greater South Information System',\n",
              "    'count': 226},\n",
              "   {'id': 'bl.cam', 'title': 'Apollo', 'count': 176},\n",
              "   {'id': 'bl.mendeley', 'title': 'Mendeley Data', 'count': 149},\n",
              "   {'id': 'cngb.cga',\n",
              "    'title': 'China National GeneBank DataBase',\n",
              "    'count': 104},\n",
              "   {'id': 'cdl.ucsd', 'title': 'UC San Diego', 'count': 92},\n",
              "   {'id': 'ubc.oc', 'title': 'Open Collections', 'count': 76},\n",
              "   {'id': 'ethz.e-coll',\n",
              "    'title': 'ETH Zürich Research Collection',\n",
              "    'count': 73}],\n",
              "  'affiliations': [{'id': 'ror.org/01ej9dk98',\n",
              "    'title': 'University of Melbourne',\n",
              "    'count': 61},\n",
              "   {'id': 'ror.org/05qbk4x57',\n",
              "    'title': 'University of Chinese Academy of Sciences',\n",
              "    'count': 59},\n",
              "   {'id': 'ror.org/04v3ywz14',\n",
              "    'title': 'China Agricultural University',\n",
              "    'count': 50},\n",
              "   {'id': 'ror.org/05ckt8b96',\n",
              "    'title': 'Ministry of Agriculture and Rural Affairs',\n",
              "    'count': 49},\n",
              "   {'id': 'ror.org/05v9jqt67',\n",
              "    'title': 'South China Agricultural University',\n",
              "    'count': 45},\n",
              "   {'id': 'ror.org/01fbgjv04',\n",
              "    'title': 'Shandong Academy of Agricultural Sciences',\n",
              "    'count': 42},\n",
              "   {'id': 'ror.org/00zxgrh39',\n",
              "    'title': 'Heilongjiang Provincial Academy of Agricultural Sciences',\n",
              "    'count': 37},\n",
              "   {'id': 'ror.org/00b30xv10',\n",
              "    'title': 'University of Pennsylvania',\n",
              "    'count': 36},\n",
              "   {'id': 'ror.org/01rkwtz72',\n",
              "    'title': 'Guangdong Academy of Agricultural Sciences',\n",
              "    'count': 36},\n",
              "   {'id': 'ror.org/0064kty71',\n",
              "    'title': 'Sun Yat-sen University',\n",
              "    'count': 34}],\n",
              "  'prefixes': [{'id': '10.6084', 'title': '10.6084', 'count': 5641},\n",
              "   {'id': '10.5281', 'title': '10.5281', 'count': 739},\n",
              "   {'id': '10.5061', 'title': '10.5061', 'count': 567},\n",
              "   {'id': '10.60692', 'title': '10.60692', 'count': 226},\n",
              "   {'id': '10.17863', 'title': '10.17863', 'count': 176},\n",
              "   {'id': '10.17632', 'title': '10.17632', 'count': 149},\n",
              "   {'id': '10.26036', 'title': '10.26036', 'count': 104},\n",
              "   {'id': '10.6075', 'title': '10.6075', 'count': 80},\n",
              "   {'id': '10.14288', 'title': '10.14288', 'count': 76},\n",
              "   {'id': '10.3929', 'title': '10.3929', 'count': 73}],\n",
              "  'certificates': [{'id': 'CoreTrustSeal',\n",
              "    'title': 'Core Trust Seal',\n",
              "    'count': 198},\n",
              "   {'id': 'DINI', 'title': 'Dini', 'count': 2},\n",
              "   {'id': 'DIN 31644', 'title': 'Din 31644', 'count': 1}],\n",
              "  'licenses': [{'id': 'cc-by-4.0', 'title': 'CC-BY-4.0', 'count': 5089},\n",
              "   {'id': 'cc0-1.0', 'title': 'CC0-1.0', 'count': 462},\n",
              "   {'id': 'cc-by-nc-sa-4.0', 'title': 'CC-BY-NC-SA-4.0', 'count': 94},\n",
              "   {'id': 'cc-by-nc-nd-4.0', 'title': 'CC-BY-NC-ND-4.0', 'count': 60},\n",
              "   {'id': 'mit', 'title': 'MIT', 'count': 47},\n",
              "   {'id': 'cc-by-nc-4.0', 'title': 'CC-BY-NC-4.0', 'count': 20},\n",
              "   {'id': 'cc-by-3.0', 'title': 'CC-BY-3.0', 'count': 10},\n",
              "   {'id': 'gpl-3.0+', 'title': 'gpl-3.0+', 'count': 9},\n",
              "   {'id': 'cc by-nc-nd 3.0', 'title': 'cc by-nc-nd 3.0', 'count': 8},\n",
              "   {'id': 'cc-by-2.0', 'title': 'CC-BY-2.0', 'count': 7}],\n",
              "  'schemaVersions': [{'id': '4', 'title': 'Schema 4', 'count': 7272},\n",
              "   {'id': '3', 'title': 'Schema 3', 'count': 1432},\n",
              "   {'id': '2.2', 'title': 'Schema 2.2', 'count': 25},\n",
              "   {'id': '4.3', 'title': 'Schema 4.3', 'count': 2}],\n",
              "  'linkChecksStatus': [{'id': '200', 'title': '200', 'count': 175},\n",
              "   {'id': '404', 'title': '404', 'count': 5},\n",
              "   {'id': '403', 'title': '403', 'count': 3},\n",
              "   {'id': '410', 'title': '410', 'count': 1}],\n",
              "  'subjects': [{'id': 'FOS: Biological sciences',\n",
              "    'title': 'Fos: Biological Sciences',\n",
              "    'count': 5412},\n",
              "   {'id': 'Genetics', 'title': 'Genetics', 'count': 4417},\n",
              "   {'id': 'Molecular Biology', 'title': 'Molecular Biology', 'count': 2038},\n",
              "   {'id': '69999 Biological Sciences not elsewhere classified',\n",
              "    'title': '69999 Biological Sciences Not Elsewhere Classified',\n",
              "    'count': 1917},\n",
              "   {'id': 'Developmental Biology',\n",
              "    'title': 'Developmental Biology',\n",
              "    'count': 1444},\n",
              "   {'id': 'Plant Biology', 'title': 'Plant Biology', 'count': 1406},\n",
              "   {'id': 'Cell Biology', 'title': 'Cell Biology', 'count': 1397},\n",
              "   {'id': 'Ecology', 'title': 'Ecology', 'count': 1380},\n",
              "   {'id': 'Microbiology', 'title': 'Microbiology', 'count': 1194},\n",
              "   {'id': 'FOS: Health sciences',\n",
              "    'title': 'Fos: Health Sciences',\n",
              "    'count': 1188}],\n",
              "  'fieldsOfScience': [{'id': 'biological_sciences',\n",
              "    'title': 'Biological sciences',\n",
              "    'count': 5412},\n",
              "   {'id': 'health_sciences', 'title': 'Health sciences', 'count': 1188},\n",
              "   {'id': 'chemical_sciences', 'title': 'Chemical sciences', 'count': 1026},\n",
              "   {'id': 'clinical_medicine', 'title': 'Clinical medicine', 'count': 968},\n",
              "   {'id': 'computer_and_information_sciences',\n",
              "    'title': 'Computer and information sciences',\n",
              "    'count': 647},\n",
              "   {'id': 'earth_and_related_environmental_sciences',\n",
              "    'title': 'Earth and related environmental sciences',\n",
              "    'count': 623},\n",
              "   {'id': 'mathematics', 'title': 'Mathematics', 'count': 242},\n",
              "   {'id': 'sociology', 'title': 'Sociology', 'count': 194},\n",
              "   {'id': 'physical_sciences', 'title': 'Physical sciences', 'count': 133},\n",
              "   {'id': 'agriculture_forestry_and_fisheries',\n",
              "    'title': 'Agriculture, forestry and fisheries',\n",
              "    'count': 18}],\n",
              "  'citations': [{'id': '2024', 'title': '2024', 'count': 17},\n",
              "   {'id': '2023', 'title': '2023', 'count': 115},\n",
              "   {'id': '2022', 'title': '2022', 'count': 234},\n",
              "   {'id': '2021', 'title': '2021', 'count': 217},\n",
              "   {'id': '2020', 'title': '2020', 'count': 98},\n",
              "   {'id': '2019', 'title': '2019', 'count': 72},\n",
              "   {'id': '2018', 'title': '2018', 'count': 52},\n",
              "   {'id': '2017', 'title': '2017', 'count': 35},\n",
              "   {'id': '2016', 'title': '2016', 'count': 51}],\n",
              "  'views': [{'id': '2023', 'title': '2023', 'count': 2765},\n",
              "   {'id': '2022', 'title': '2022', 'count': 3957},\n",
              "   {'id': '2021', 'title': '2021', 'count': 6279},\n",
              "   {'id': '2020', 'title': '2020', 'count': 10377},\n",
              "   {'id': '2019', 'title': '2019', 'count': 27073},\n",
              "   {'id': '2018', 'title': '2018', 'count': 16576},\n",
              "   {'id': '2017', 'title': '2017', 'count': 13498},\n",
              "   {'id': '2016', 'title': '2016', 'count': 17621}],\n",
              "  'downloads': [{'id': '2023', 'title': '2023', 'count': 204},\n",
              "   {'id': '2022', 'title': '2022', 'count': 401},\n",
              "   {'id': '2021', 'title': '2021', 'count': 991},\n",
              "   {'id': '2020', 'title': '2020', 'count': 1224},\n",
              "   {'id': '2019', 'title': '2019', 'count': 4160},\n",
              "   {'id': '2018', 'title': '2018', 'count': 2963},\n",
              "   {'id': '2017', 'title': '2017', 'count': 2707},\n",
              "   {'id': '2016', 'title': '2016', 'count': 2822}]},\n",
              " 'links': {'self': 'https://api.datacite.org/dois/?query=transcriptomes&page%5Bsize%5D=10',\n",
              "  'next': 'https://api.datacite.org/dois?page%5Bnumber%5D=2&page%5Bsize%5D=10&query=transcriptomes'}}"
            ]
          },
          "metadata": {},
          "execution_count": 11
        }
      ],
      "source": [
        "results"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "u3AWMg9ZaDBe"
      },
      "source": [
        "To look at the dois in the resonse we will need to access the `data` key of the results dictionary"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "collapsed": true,
        "id": "9eFqoyknaAXa"
      },
      "outputs": [],
      "source": [
        "dois = results['data']"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "o9sAn6P1kStI",
        "outputId": "0f826566-9bec-45af-b7da-f4c61354f392"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "10"
            ]
          },
          "metadata": {},
          "execution_count": 14
        }
      ],
      "source": [
        "len(dois)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "uYDRWFIqW4FB"
      },
      "source": [
        "Using the above method you can change the query, or use any of the search parameters outlined in the [API reference](https://support.datacite.org/reference/get_dois)\n",
        "\n",
        "A you can even increase the number of rows returned up to 1000\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "WGCLbW7Dd7lc"
      },
      "outputs": [],
      "source": [
        "rows = 1000\n",
        "params = {\n",
        "        'query': my_query,\n",
        "        'page[size]': rows\n",
        "    }\n",
        "\n",
        "url = DATACITE_DOI_API_URL\n",
        "response = requests.get(url, params)\n",
        "if response.ok:\n",
        "  results = response.json()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "hILcXNiGaRLh"
      },
      "source": [
        "### Pagination to retrieve over 1000 results\n",
        "\n",
        "Note: using page numbers we can only retrieve up to 10,000 results"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3ZrZglsZbKGB"
      },
      "source": [
        "The row max is limited to 1000 but we can request mutlitple pages of dois.  First, let's look at the total number of results, and the total number of pages.  These are located in the `meta` sub-dictionary"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9VHkF6SQbZNh",
        "outputId": "3f8c66ac-dddc-4490-f687-5ba797e98123"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "8946"
            ]
          },
          "metadata": {},
          "execution_count": 16
        }
      ],
      "source": [
        "results['meta']['total'] # Total number of results\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aY8vpf6UcgiB",
        "outputId": "3af3e0f5-0364-472a-cbae-c6b8dbd21477"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "9"
            ]
          },
          "metadata": {},
          "execution_count": 17
        }
      ],
      "source": [
        "results['meta']['totalPages'] # Total number of pages"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Imwhz2oXfK65"
      },
      "source": [
        "We will need to add the `page[number]` to our list of parameters"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "VBjUyzMdfgPL"
      },
      "outputs": [],
      "source": [
        "params = {\n",
        "        'query': my_query,\n",
        "        'page[size]': rows,\n",
        "        'page[number]': 1\n",
        "    }\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "mn-K2Q7xfiXr"
      },
      "source": [
        "Lets put it all together.  We will\n",
        "\n",
        "1.   Perform an initial search\n",
        "2.   Get the total number of pages\n",
        "3.   Create a for loop to run the search on each page\n",
        "\n",
        "Meanwhile we will collect all dois returned to the list `all_dois`\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "url = DATACITE_DOI_API_URL\n",
        "\n",
        "params = {\n",
        "        'query': my_query,\n",
        "        'page[size]': 0,\n",
        "        'page[number]': 1\n",
        "    }\n",
        "\n",
        "response = requests.get(url, params)\n",
        "response.json()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "V1te9u63fol7",
        "outputId": "fbc844bd-cb54-4b4c-8173-6a4a3e59dbc4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'data': [],\n",
              " 'meta': {'total': 8948,\n",
              "  'totalPages': 0,\n",
              "  'page': 1,\n",
              "  'states': [{'id': 'findable', 'title': 'Findable', 'count': 8948}],\n",
              "  'resourceTypes': [{'id': 'dataset', 'title': 'Dataset', 'count': 3434},\n",
              "   {'id': 'collection', 'title': 'Collection', 'count': 2293},\n",
              "   {'id': 'text', 'title': 'Text', 'count': 2019},\n",
              "   {'id': 'image', 'title': 'Image', 'count': 469},\n",
              "   {'id': 'journal-article', 'title': 'Journal Article', 'count': 186},\n",
              "   {'id': 'software', 'title': 'Software', 'count': 160},\n",
              "   {'id': 'other', 'title': 'Other', 'count': 124},\n",
              "   {'id': 'dissertation', 'title': 'Dissertation', 'count': 69},\n",
              "   {'id': 'audiovisual', 'title': 'Audiovisual', 'count': 62},\n",
              "   {'id': 'preprint', 'title': 'Preprint', 'count': 21},\n",
              "   {'id': 'workflow', 'title': 'Workflow', 'count': 14},\n",
              "   {'id': 'peer-review', 'title': 'Peer Review', 'count': 10},\n",
              "   {'id': 'output-management-plan',\n",
              "    'title': 'Output Management Plan',\n",
              "    'count': 9},\n",
              "   {'id': 'project', 'title': 'Project', 'count': 5},\n",
              "   {'id': 'conference-paper', 'title': 'Conference Paper', 'count': 3},\n",
              "   {'id': 'data-paper', 'title': 'Data Paper', 'count': 3},\n",
              "   {'id': 'report', 'title': 'Report', 'count': 2},\n",
              "   {'id': 'interactive-resource',\n",
              "    'title': 'Interactive Resource',\n",
              "    'count': 1}],\n",
              "  'created': [{'id': '2024', 'title': '2024', 'count': 634},\n",
              "   {'id': '2023', 'title': '2023', 'count': 850},\n",
              "   {'id': '2022', 'title': '2022', 'count': 1156},\n",
              "   {'id': '2021', 'title': '2021', 'count': 1171},\n",
              "   {'id': '2020', 'title': '2020', 'count': 1087},\n",
              "   {'id': '2019', 'title': '2019', 'count': 1218},\n",
              "   {'id': '2018', 'title': '2018', 'count': 844},\n",
              "   {'id': '2017', 'title': '2017', 'count': 1068},\n",
              "   {'id': '2016', 'title': '2016', 'count': 619},\n",
              "   {'id': '2015', 'title': '2015', 'count': 127}],\n",
              "  'published': [{'id': '2024', 'title': '2024', 'count': 456},\n",
              "   {'id': '2023', 'title': '2023', 'count': 867},\n",
              "   {'id': '2022', 'title': '2022', 'count': 1166},\n",
              "   {'id': '2021', 'title': '2021', 'count': 1091},\n",
              "   {'id': '2020', 'title': '2020', 'count': 1035},\n",
              "   {'id': '2019', 'title': '2019', 'count': 1224},\n",
              "   {'id': '2018', 'title': '2018', 'count': 795},\n",
              "   {'id': '2017', 'title': '2017', 'count': 762},\n",
              "   {'id': '2016', 'title': '2016', 'count': 659},\n",
              "   {'id': '2015', 'title': '2015', 'count': 449},\n",
              "   {'id': '2014', 'title': '2014', 'count': 107},\n",
              "   {'id': '2013', 'title': '2013', 'count': 82},\n",
              "   {'id': '2012', 'title': '2012', 'count': 69},\n",
              "   {'id': '2011', 'title': '2011', 'count': 142},\n",
              "   {'id': '2010', 'title': '2010', 'count': 14}],\n",
              "  'registered': [{'id': '2024', 'title': '2024', 'count': 637},\n",
              "   {'id': '2023', 'title': '2023', 'count': 849},\n",
              "   {'id': '2022', 'title': '2022', 'count': 1158},\n",
              "   {'id': '2021', 'title': '2021', 'count': 1169},\n",
              "   {'id': '2020', 'title': '2020', 'count': 1085},\n",
              "   {'id': '2019', 'title': '2019', 'count': 1218},\n",
              "   {'id': '2018', 'title': '2018', 'count': 843},\n",
              "   {'id': '2017', 'title': '2017', 'count': 1067},\n",
              "   {'id': '2016', 'title': '2016', 'count': 619},\n",
              "   {'id': '2015', 'title': '2015', 'count': 127}],\n",
              "  'providers': [{'id': 'otjm', 'title': 'Figshare Internal', 'count': 5641},\n",
              "   {'id': 'cern',\n",
              "    'title': 'CERN - European Organization for Nuclear Research',\n",
              "    'count': 739},\n",
              "   {'id': 'dryad', 'title': 'Dryad', 'count': 579},\n",
              "   {'id': 'iapx',\n",
              "    'title': 'Organisation of Southern Cooperation (OSC)',\n",
              "    'count': 228},\n",
              "   {'id': 'lpsw', 'title': 'University of Cambridge', 'count': 176},\n",
              "   {'id': 'cngb', 'title': 'China National GeneBank', 'count': 154},\n",
              "   {'id': 'elsevier', 'title': 'Elsevier (Mendeley Data)', 'count': 149},\n",
              "   {'id': 'cizs', 'title': 'UC San Diego', 'count': 92},\n",
              "   {'id': 'ubc', 'title': 'University of British Columbia', 'count': 76},\n",
              "   {'id': 'stdp', 'title': 'ETH Zurich', 'count': 73}],\n",
              "  'clients': [{'id': 'figshare.ars',\n",
              "    'title': 'figshare Academic Research System',\n",
              "    'count': 5641},\n",
              "   {'id': 'cern.zenodo', 'title': 'Zenodo', 'count': 739},\n",
              "   {'id': 'dryad.dryad', 'title': 'DRYAD', 'count': 579},\n",
              "   {'id': 'iapx.bsycxq',\n",
              "    'title': 'Greater South Information System',\n",
              "    'count': 228},\n",
              "   {'id': 'bl.cam', 'title': 'Apollo', 'count': 176},\n",
              "   {'id': 'bl.mendeley', 'title': 'Mendeley Data', 'count': 149},\n",
              "   {'id': 'cngb.cga',\n",
              "    'title': 'China National GeneBank DataBase',\n",
              "    'count': 104},\n",
              "   {'id': 'cdl.ucsd', 'title': 'UC San Diego', 'count': 92},\n",
              "   {'id': 'ubc.oc', 'title': 'Open Collections', 'count': 76},\n",
              "   {'id': 'ethz.e-coll',\n",
              "    'title': 'ETH Zürich Research Collection',\n",
              "    'count': 73}],\n",
              "  'affiliations': [{'id': 'ror.org/01ej9dk98',\n",
              "    'title': 'University of Melbourne',\n",
              "    'count': 61},\n",
              "   {'id': 'ror.org/05qbk4x57',\n",
              "    'title': 'University of Chinese Academy of Sciences',\n",
              "    'count': 59},\n",
              "   {'id': 'ror.org/04v3ywz14',\n",
              "    'title': 'China Agricultural University',\n",
              "    'count': 50},\n",
              "   {'id': 'ror.org/05ckt8b96',\n",
              "    'title': 'Ministry of Agriculture and Rural Affairs',\n",
              "    'count': 49},\n",
              "   {'id': 'ror.org/05v9jqt67',\n",
              "    'title': 'South China Agricultural University',\n",
              "    'count': 45},\n",
              "   {'id': 'ror.org/01fbgjv04',\n",
              "    'title': 'Shandong Academy of Agricultural Sciences',\n",
              "    'count': 42},\n",
              "   {'id': 'ror.org/00zxgrh39',\n",
              "    'title': 'Heilongjiang Provincial Academy of Agricultural Sciences',\n",
              "    'count': 37},\n",
              "   {'id': 'ror.org/00b30xv10',\n",
              "    'title': 'University of Pennsylvania',\n",
              "    'count': 36},\n",
              "   {'id': 'ror.org/01rkwtz72',\n",
              "    'title': 'Guangdong Academy of Agricultural Sciences',\n",
              "    'count': 36},\n",
              "   {'id': 'ror.org/0064kty71',\n",
              "    'title': 'Sun Yat-sen University',\n",
              "    'count': 34}],\n",
              "  'prefixes': [{'id': '10.6084', 'title': '10.6084', 'count': 5641},\n",
              "   {'id': '10.5281', 'title': '10.5281', 'count': 739},\n",
              "   {'id': '10.5061', 'title': '10.5061', 'count': 567},\n",
              "   {'id': '10.60692', 'title': '10.60692', 'count': 228},\n",
              "   {'id': '10.17863', 'title': '10.17863', 'count': 176},\n",
              "   {'id': '10.17632', 'title': '10.17632', 'count': 149},\n",
              "   {'id': '10.26036', 'title': '10.26036', 'count': 104},\n",
              "   {'id': '10.6075', 'title': '10.6075', 'count': 80},\n",
              "   {'id': '10.14288', 'title': '10.14288', 'count': 76},\n",
              "   {'id': '10.3929', 'title': '10.3929', 'count': 73}],\n",
              "  'certificates': [{'id': 'CoreTrustSeal',\n",
              "    'title': 'Core Trust Seal',\n",
              "    'count': 198},\n",
              "   {'id': 'DINI', 'title': 'Dini', 'count': 2},\n",
              "   {'id': 'DIN 31644', 'title': 'Din 31644', 'count': 1}],\n",
              "  'licenses': [{'id': 'cc-by-4.0', 'title': 'CC-BY-4.0', 'count': 5089},\n",
              "   {'id': 'cc0-1.0', 'title': 'CC0-1.0', 'count': 462},\n",
              "   {'id': 'cc-by-nc-sa-4.0', 'title': 'CC-BY-NC-SA-4.0', 'count': 94},\n",
              "   {'id': 'cc-by-nc-nd-4.0', 'title': 'CC-BY-NC-ND-4.0', 'count': 60},\n",
              "   {'id': 'mit', 'title': 'MIT', 'count': 47},\n",
              "   {'id': 'cc-by-nc-4.0', 'title': 'CC-BY-NC-4.0', 'count': 20},\n",
              "   {'id': 'cc-by-3.0', 'title': 'CC-BY-3.0', 'count': 10},\n",
              "   {'id': 'gpl-3.0+', 'title': 'gpl-3.0+', 'count': 9},\n",
              "   {'id': 'cc by-nc-nd 3.0', 'title': 'cc by-nc-nd 3.0', 'count': 8},\n",
              "   {'id': 'cc-by-2.0', 'title': 'CC-BY-2.0', 'count': 7}],\n",
              "  'schemaVersions': [{'id': '4', 'title': 'Schema 4', 'count': 7274},\n",
              "   {'id': '3', 'title': 'Schema 3', 'count': 1432},\n",
              "   {'id': '2.2', 'title': 'Schema 2.2', 'count': 25},\n",
              "   {'id': '4.3', 'title': 'Schema 4.3', 'count': 2}],\n",
              "  'linkChecksStatus': [{'id': '200', 'title': '200', 'count': 175},\n",
              "   {'id': '404', 'title': '404', 'count': 5},\n",
              "   {'id': '403', 'title': '403', 'count': 3},\n",
              "   {'id': '410', 'title': '410', 'count': 1}],\n",
              "  'subjects': [{'id': 'FOS: Biological sciences',\n",
              "    'title': 'Fos: Biological Sciences',\n",
              "    'count': 5414},\n",
              "   {'id': 'Genetics', 'title': 'Genetics', 'count': 4419},\n",
              "   {'id': 'Molecular Biology', 'title': 'Molecular Biology', 'count': 2040},\n",
              "   {'id': '69999 Biological Sciences not elsewhere classified',\n",
              "    'title': '69999 Biological Sciences Not Elsewhere Classified',\n",
              "    'count': 1917},\n",
              "   {'id': 'Developmental Biology',\n",
              "    'title': 'Developmental Biology',\n",
              "    'count': 1444},\n",
              "   {'id': 'Plant Biology', 'title': 'Plant Biology', 'count': 1406},\n",
              "   {'id': 'Cell Biology', 'title': 'Cell Biology', 'count': 1397},\n",
              "   {'id': 'Ecology', 'title': 'Ecology', 'count': 1380},\n",
              "   {'id': 'Microbiology', 'title': 'Microbiology', 'count': 1194},\n",
              "   {'id': 'FOS: Health sciences',\n",
              "    'title': 'Fos: Health Sciences',\n",
              "    'count': 1188}],\n",
              "  'fieldsOfScience': [{'id': 'biological_sciences',\n",
              "    'title': 'Biological sciences',\n",
              "    'count': 5414},\n",
              "   {'id': 'health_sciences', 'title': 'Health sciences', 'count': 1188},\n",
              "   {'id': 'chemical_sciences', 'title': 'Chemical sciences', 'count': 1026},\n",
              "   {'id': 'clinical_medicine', 'title': 'Clinical medicine', 'count': 968},\n",
              "   {'id': 'computer_and_information_sciences',\n",
              "    'title': 'Computer and information sciences',\n",
              "    'count': 649},\n",
              "   {'id': 'earth_and_related_environmental_sciences',\n",
              "    'title': 'Earth and related environmental sciences',\n",
              "    'count': 623},\n",
              "   {'id': 'mathematics', 'title': 'Mathematics', 'count': 242},\n",
              "   {'id': 'sociology', 'title': 'Sociology', 'count': 194},\n",
              "   {'id': 'physical_sciences', 'title': 'Physical sciences', 'count': 133},\n",
              "   {'id': 'agriculture_forestry_and_fisheries',\n",
              "    'title': 'Agriculture, forestry and fisheries',\n",
              "    'count': 18}],\n",
              "  'citations': [{'id': '2024', 'title': '2024', 'count': 17},\n",
              "   {'id': '2023', 'title': '2023', 'count': 115},\n",
              "   {'id': '2022', 'title': '2022', 'count': 234},\n",
              "   {'id': '2021', 'title': '2021', 'count': 217},\n",
              "   {'id': '2020', 'title': '2020', 'count': 98},\n",
              "   {'id': '2019', 'title': '2019', 'count': 72},\n",
              "   {'id': '2018', 'title': '2018', 'count': 52},\n",
              "   {'id': '2017', 'title': '2017', 'count': 35},\n",
              "   {'id': '2016', 'title': '2016', 'count': 51}],\n",
              "  'views': [{'id': '2023', 'title': '2023', 'count': 2765},\n",
              "   {'id': '2022', 'title': '2022', 'count': 3957},\n",
              "   {'id': '2021', 'title': '2021', 'count': 6279},\n",
              "   {'id': '2020', 'title': '2020', 'count': 10377},\n",
              "   {'id': '2019', 'title': '2019', 'count': 27073},\n",
              "   {'id': '2018', 'title': '2018', 'count': 16576},\n",
              "   {'id': '2017', 'title': '2017', 'count': 13498},\n",
              "   {'id': '2016', 'title': '2016', 'count': 17621}],\n",
              "  'downloads': [{'id': '2023', 'title': '2023', 'count': 204},\n",
              "   {'id': '2022', 'title': '2022', 'count': 401},\n",
              "   {'id': '2021', 'title': '2021', 'count': 991},\n",
              "   {'id': '2020', 'title': '2020', 'count': 1224},\n",
              "   {'id': '2019', 'title': '2019', 'count': 4160},\n",
              "   {'id': '2018', 'title': '2018', 'count': 2963},\n",
              "   {'id': '2017', 'title': '2017', 'count': 2707},\n",
              "   {'id': '2016', 'title': '2016', 'count': 2822}]},\n",
              " 'links': {'self': 'https://api.datacite.org/dois/?query=transcriptomes&page%5Bsize%5D=0&page%5Bnumber%5D=1'}}"
            ]
          },
          "metadata": {},
          "execution_count": 20
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "nqeX8G5ocwTu"
      },
      "outputs": [],
      "source": [
        "# Set up lists to collect the results\n",
        "all_dois = []\n",
        "rows = 1000\n",
        "my_query = 'transcriptomes'\n",
        "url = DATACITE_DOI_API_URL\n",
        "\n",
        "# Initial Search\n",
        "\n",
        "params = {\n",
        "        'query': my_query,\n",
        "        'page[size]': rows,\n",
        "        'page[number]': 1\n",
        "    }\n",
        "\n",
        "response = requests.get(url, params)\n",
        "if response.ok:\n",
        "  results = response.json()\n",
        "  all_dois.extend(results['data'])\n",
        "\n",
        "# Get the total number of pages\n",
        "total_number_of_pages = results['meta']['totalPages'] # Total number of pages\n",
        "\n",
        "# Loop through all the remaining pages\n",
        "for page in range(2, total_number_of_pages + 1):\n",
        "  # Update our parameters\n",
        "\n",
        "  params = {\n",
        "        'query': my_query,\n",
        "        'page[size]': rows,\n",
        "        'page[number]': page\n",
        "  }\n",
        "\n",
        "  response = requests.get(url, params)\n",
        "  if response.ok:\n",
        "    results = response.json()\n",
        "    all_dois.extend(results['data'])\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "jYPCaYungpl-"
      },
      "source": [
        "Let's check the size of the all_dois list"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "JxzT-wLNgtK2",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4f790381-e60f-4300-a1e1-10065e29eba9"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "8948"
            ]
          },
          "metadata": {},
          "execution_count": 21
        }
      ],
      "source": [
        "len(all_dois)\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "pZgKney38Qrp"
      },
      "source": [
        "Preview the results using the pandas library"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "8fSqcD3m8QIb",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 424
        },
        "outputId": "c621f201-c10a-4f53-92e3-21dc530b4151"
      },
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "                                    id  type  \\\n",
              "0                 10.60692/pgtps-9b067  dois   \n",
              "1                 10.60692/85e10-n3c87  dois   \n",
              "2                     10.7488/era/4551  dois   \n",
              "3                 10.60692/k1t8q-38c08  dois   \n",
              "4                 10.60692/7mnfs-10y79  dois   \n",
              "...                                ...   ...   \n",
              "8943     10.6084/m9.figshare.c.5022629  dois   \n",
              "8944            10.5281/zenodo.3885088  dois   \n",
              "8945  10.6084/m9.figshare.c.4958501.v1  dois   \n",
              "8946            10.5281/zenodo.1493871  dois   \n",
              "8947            10.5281/zenodo.3731264  dois   \n",
              "\n",
              "                                             attributes  \\\n",
              "0     {'doi': '10.60692/pgtps-9b067', 'identifiers':...   \n",
              "1     {'doi': '10.60692/85e10-n3c87', 'identifiers':...   \n",
              "2     {'doi': '10.7488/era/4551', 'identifiers': [{'...   \n",
              "3     {'doi': '10.60692/k1t8q-38c08', 'identifiers':...   \n",
              "4     {'doi': '10.60692/7mnfs-10y79', 'identifiers':...   \n",
              "...                                                 ...   \n",
              "8943  {'doi': '10.6084/m9.figshare.c.5022629', 'iden...   \n",
              "8944  {'doi': '10.5281/zenodo.3885088', 'identifiers...   \n",
              "8945  {'doi': '10.6084/m9.figshare.c.4958501.v1', 'i...   \n",
              "8946  {'doi': '10.5281/zenodo.1493871', 'identifiers...   \n",
              "8947  {'doi': '10.5281/zenodo.3731264', 'identifiers...   \n",
              "\n",
              "                                          relationships  \n",
              "0     {'client': {'data': {'id': 'iapx.bsycxq', 'typ...  \n",
              "1     {'client': {'data': {'id': 'iapx.bsycxq', 'typ...  \n",
              "2     {'client': {'data': {'id': 'bl.ed', 'type': 'c...  \n",
              "3     {'client': {'data': {'id': 'iapx.bsycxq', 'typ...  \n",
              "4     {'client': {'data': {'id': 'iapx.bsycxq', 'typ...  \n",
              "...                                                 ...  \n",
              "8943  {'client': {'data': {'id': 'figshare.ars', 'ty...  \n",
              "8944  {'client': {'data': {'id': 'cern.zenodo', 'typ...  \n",
              "8945  {'client': {'data': {'id': 'figshare.ars', 'ty...  \n",
              "8946  {'client': {'data': {'id': 'cern.zenodo', 'typ...  \n",
              "8947  {'client': {'data': {'id': 'cern.zenodo', 'typ...  \n",
              "\n",
              "[8948 rows x 4 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-d944d512-0733-400b-b68d-3dd680f8d69b\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>id</th>\n",
              "      <th>type</th>\n",
              "      <th>attributes</th>\n",
              "      <th>relationships</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>10.60692/pgtps-9b067</td>\n",
              "      <td>dois</td>\n",
              "      <td>{'doi': '10.60692/pgtps-9b067', 'identifiers':...</td>\n",
              "      <td>{'client': {'data': {'id': 'iapx.bsycxq', 'typ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>10.60692/85e10-n3c87</td>\n",
              "      <td>dois</td>\n",
              "      <td>{'doi': '10.60692/85e10-n3c87', 'identifiers':...</td>\n",
              "      <td>{'client': {'data': {'id': 'iapx.bsycxq', 'typ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>10.7488/era/4551</td>\n",
              "      <td>dois</td>\n",
              "      <td>{'doi': '10.7488/era/4551', 'identifiers': [{'...</td>\n",
              "      <td>{'client': {'data': {'id': 'bl.ed', 'type': 'c...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>10.60692/k1t8q-38c08</td>\n",
              "      <td>dois</td>\n",
              "      <td>{'doi': '10.60692/k1t8q-38c08', 'identifiers':...</td>\n",
              "      <td>{'client': {'data': {'id': 'iapx.bsycxq', 'typ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>10.60692/7mnfs-10y79</td>\n",
              "      <td>dois</td>\n",
              "      <td>{'doi': '10.60692/7mnfs-10y79', 'identifiers':...</td>\n",
              "      <td>{'client': {'data': {'id': 'iapx.bsycxq', 'typ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8943</th>\n",
              "      <td>10.6084/m9.figshare.c.5022629</td>\n",
              "      <td>dois</td>\n",
              "      <td>{'doi': '10.6084/m9.figshare.c.5022629', 'iden...</td>\n",
              "      <td>{'client': {'data': {'id': 'figshare.ars', 'ty...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8944</th>\n",
              "      <td>10.5281/zenodo.3885088</td>\n",
              "      <td>dois</td>\n",
              "      <td>{'doi': '10.5281/zenodo.3885088', 'identifiers...</td>\n",
              "      <td>{'client': {'data': {'id': 'cern.zenodo', 'typ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8945</th>\n",
              "      <td>10.6084/m9.figshare.c.4958501.v1</td>\n",
              "      <td>dois</td>\n",
              "      <td>{'doi': '10.6084/m9.figshare.c.4958501.v1', 'i...</td>\n",
              "      <td>{'client': {'data': {'id': 'figshare.ars', 'ty...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8946</th>\n",
              "      <td>10.5281/zenodo.1493871</td>\n",
              "      <td>dois</td>\n",
              "      <td>{'doi': '10.5281/zenodo.1493871', 'identifiers...</td>\n",
              "      <td>{'client': {'data': {'id': 'cern.zenodo', 'typ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8947</th>\n",
              "      <td>10.5281/zenodo.3731264</td>\n",
              "      <td>dois</td>\n",
              "      <td>{'doi': '10.5281/zenodo.3731264', 'identifiers...</td>\n",
              "      <td>{'client': {'data': {'id': 'cern.zenodo', 'typ...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>8948 rows × 4 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-d944d512-0733-400b-b68d-3dd680f8d69b')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-d944d512-0733-400b-b68d-3dd680f8d69b button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-d944d512-0733-400b-b68d-3dd680f8d69b');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-4c306ef4-b1a0-49e6-9a67-b674c4df011c\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-4c306ef4-b1a0-49e6-9a67-b674c4df011c')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-4c306ef4-b1a0-49e6-9a67-b674c4df011c button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df",
              "summary": "{\n  \"name\": \"df\",\n  \"rows\": 8948,\n  \"fields\": [\n    {\n      \"column\": \"id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8948,\n        \"samples\": [\n          \"10.6084/m9.figshare.c.3595919_d1.v1\",\n          \"10.7912/c24s44\",\n          \"10.6084/m9.figshare.13071086.v1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"type\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"dois\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"attributes\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"relationships\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {}
        }
      ],
      "source": [
        "import pandas as pd\n",
        "df = pd.DataFrame(all_dois)\n",
        "display(df)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Convert the results to different file formats and save them"
      ],
      "metadata": {
        "id": "CdGYgSUfC4ao"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "df.to_json(\"search_results.json\", indent=2)\n",
        "df.to_csv(\"search_results.csv\", index=False)"
      ],
      "metadata": {
        "id": "uL72JL8IDCZi"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## (Optional) Using cursors for paginating through results"
      ],
      "metadata": {
        "id": "CEK89ujS8MP5"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "As highlighted in the [API reference](https://support.datacite.org/docs/pagination#method-2-cursor), you can use the parameter `page[cursor]=1` in the initial search and use the `links[next]` value in the results from there on out."
      ],
      "metadata": {
        "id": "ilbD_vDeE4er"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# Set up lists to collect the results\n",
        "all_dois = []\n",
        "rows = 1000\n",
        "my_query = 'transcriptomes'\n",
        "url = DATACITE_DOI_API_URL\n",
        "\n",
        "# Initial Search\n",
        "\n",
        "params = {\n",
        "        'query': my_query,\n",
        "        'page[size]': rows\n",
        "}\n",
        "while url:\n",
        "    response = requests.get(url, params=params)\n",
        "    if response.status_code == 200:\n",
        "      params.pop('page[cursor]', None) # Remove page[cursor] from the params if it exists\n",
        "      results = response.json()\n",
        "      if results and 'data' in results:\n",
        "        all_dois.extend(results['data'])\n",
        "        url = results['links'].get('next', None) # Update the URL to the next page, if any\n",
        "      else:\n",
        "        break\n",
        "    else:\n",
        "      print(f\"Error: {response.status_code}\")\n",
        "      break"
      ],
      "metadata": {
        "id": "bXUlIQBZ8gAm"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(all_dois)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8buBljTgECvH",
        "outputId": "b53ad503-4455-4572-9be5-6f24bdcc9975"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "8851"
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}